diff --git a/django/db/models/query.py b/django/db/models/query.py
index b48d0df9c0..d1d1587b07 100644
--- a/django/db/models/query.py
+++ b/django/db/models/query.py
@@ -211,6 +211,11 @@ class QuerySet:
     @query.setter
     def query(self, value):
         self._query = value
+        # Update the iterable class to match the query
+        if value.values_select:
+            self._iterable_class = ValuesIterable
+        else:
+            self._iterable_class = ModelIterable
 
     def as_manager(cls):
         # Address the circular dependency between `Queryset` and `Manager`.
