diff --git a/django/db/models/sql/query.py b/django/db/models/sql/query.py
index ce18098fd2..7e1cb71ba9 100644
--- a/django/db/models/sql/query.py
+++ b/django/db/models/sql/query.py
@@ -1122,11 +1122,11 @@ class Query(BaseExpression):
                 for v in value:
                     self.check_query_object_type(v, opts, field)
 
     def check_filterable(self, expression):
         """Raise an error if expression cannot be used in a WHERE clause."""
-        if not getattr(expression, 'filterable', True):
+        if isinstance(expression, BaseExpression) and not getattr(expression, 'filterable', True):
             raise NotSupportedError(
                 expression.__class__.__name__ + ' is disallowed in the filter '
                 'clause.'
             )
         if hasattr(expression, 'get_source_expressions'):
