diff --git a/django/contrib/messages/storage/base.py b/django/contrib/messages/storage/base.py
index 01422066a6..fba7264bee 100644
--- a/django/contrib/messages/storage/base.py
+++ b/django/contrib/messages/storage/base.py
@@ -1,11 +1,23 @@
 from django.conf import settings
 from django.contrib.messages import constants, utils
+from django.core.signals import setting_changed
+from django.dispatch import receiver
 
 LEVEL_TAGS = utils.get_level_tags()
 
 
+@receiver(setting_changed)
+def update_level_tags(sender, setting, **kwargs):
+    """
+    Update LEVEL_TAGS when MESSAGE_TAGS setting changes.
+    """
+    if setting == 'MESSAGE_TAGS':
+        global LEVEL_TAGS
+        LEVEL_TAGS = utils.get_level_tags()
+
+
 class Message:
     """
     Represent an actual message that can be stored in any of the supported
     storage classes (typically session- or cookie-based) and rendered in a view
     or template.
