diff --git a/src/flask/blueprints.py b/src/flask/blueprints.py
index eb50585d..f15bfcf3 100644
--- a/src/flask/blueprints.py
+++ b/src/flask/blueprints.py
@@ -188,10 +188,13 @@ class Blueprint(Scaffold):
             static_url_path=static_url_path,
             template_folder=template_folder,
             root_path=root_path,
         )
 
+        if not name.strip():
+            raise ValueError("'name' may not be empty or contain only whitespace.")
+
         if "." in name:
             raise ValueError("'name' may not contain a dot '.' character.")
 
         self.name = name
         self.url_prefix = url_prefix
