diff --git a/sklearn/multioutput.py b/sklearn/multioutput.py
index 463b72d40..3ba0e1ffd 100644
--- a/sklearn/multioutput.py
+++ b/sklearn/multioutput.py
@@ -316,17 +316,34 @@ class MultiOutputClassifier(MultiOutputEstimator, ClassifierMixin):
         ``-1`` means using all processors. See :term:`Glossary <n_jobs>`
         for more details.
 
     Attributes
     ----------
+    classes_ : list of arrays
+        Class labels for each output. Each array contains the unique
+        class labels for the corresponding output.
+        
     estimators_ : list of ``n_output`` estimators
         Estimators used for predictions.
     """
 
     def __init__(self, estimator, n_jobs=None):
         super().__init__(estimator, n_jobs)
 
+    @property
+    def classes_(self):
+        """Class labels for each output.
+        
+        Returns
+        -------
+        classes_ : list of arrays
+            Class labels for each output. Each array contains the unique
+            class labels for the corresponding output.
+        """
+        check_is_fitted(self)
+        return [estimator.classes_ for estimator in self.estimators_]
+
     def predict_proba(self, X):
         """Probability estimates.
         Returns prediction probabilities for each class of each output.
 
         This method will raise a ``ValueError`` if any of the
