diff --git b/sphinx/writers/latex.py a/sphinx/writers/latex.py
index b14e663e7..e0c7d56f8 100644
--- b/sphinx/writers/latex.py
+++ a/sphinx/writers/latex.py
@@ -1707,16 +1707,16 @@ class LaTeXTranslator(SphinxTranslator):
         opts = self.config.highlight_options.get(lang, {})
         hlcode = self.highlighter.highlight_block(
             node.astext(), lang, opts=opts, location=node)
         # TODO: Use nowrap option once LaTeX formatter supports it
         # https://github.com/pygments/pygments/pull/1343
-        hlcode = hlcode.replace(r'\begin{Verbatim}[commandchars=\\\{\}]' + '\n',
+        hlcode = hlcode.replace(r'\begin{Verbatim}[commandchars=\\\{\}]',
                                 r'\sphinxcode{\sphinxupquote{')
         # get consistent trailer
-        hlcode = hlcode.replace('\n' + r'\end{Verbatim}', '}}')
-        hlcode = hlcode.rstrip()  # remove any trailing whitespace
+        hlcode = hlcode.rstrip()[:-14]  # strip \end{Verbatim}
         self.body.append(hlcode)
+        self.body.append('}}')
         raise nodes.SkipNode
 
     def depart_literal(self, node: Element) -> None:
         self.body.append('}}')
 
