diff --git a/sympy/polys/polytools.py b/sympy/polys/polytools.py
index d5ac051f63..1acaa10988 100644
--- a/sympy/polys/polytools.py
+++ b/sympy/polys/polytools.py
@@ -104,10 +104,11 @@ class Poly(Expr):
 
     __slots__ = ['rep', 'gens']
 
     is_commutative = True
     is_Poly = True
+    _op_priority = 10.1  # higher than Expr (10.0) to ensure Poly.__rmul__ is called
 
     def __new__(cls, rep, *gens, **args):
         """Create a new polynomial instance out of something useful. """
         opt = options.build_options(gens, args)
 
