diff --git a/sympy/codegen/ast.py b/sympy/codegen/ast.py
index a02714b174..db19dbfc09 100644
--- a/sympy/codegen/ast.py
+++ b/sympy/codegen/ast.py
@@ -895,7 +895,6 @@ class String(Token):
 
     """
     __slots__ = ('text',)
-    not_in_args = ['text']
     is_Atom = True
 
     @classmethod
@@ -906,6 +905,10 @@ def _construct_text(cls, text):
 
     def _sympystr(self, printer, *args, **kwargs):
         return self.text
+    
+    @property
+    def args(self):
+        return (self.text,)
 
 
 class QuotedString(String):
