diff --git a/django/db/models/deletion.py b/django/db/models/deletion.py
index 3a462f1c4c..6eae85f559 100644
--- a/django/db/models/deletion.py
+++ b/django/db/models/deletion.py
@@ -232,9 +232,11 @@ class Collector:
         """
         Get a QuerySet of objects related to `objs` via the relation `related`.
         """
+        # Only select the pk field to avoid loading all field values, which
+        # might include invalid unicode data that causes errors on load
         return related.related_model._base_manager.using(self.using).filter(
             **{"%s__in" % related.field.name: objs}
-        )
+        ).only('pk')
 
     def instances_with_model(self):
         for model, instances in self.data.items():

