diff --git a/django/db/models/base.py b/django/db/models/base.py
index e3b14a41a0..ecc36d3fda 100644
--- a/django/db/models/base.py
+++ b/django/db/models/base.py
@@ -848,11 +848,15 @@ class Model(metaclass=ModelBase):
             raise ValueError("Cannot force an update in save() with no primary key.")
         updated = False
         # Skip an UPDATE when adding an instance and primary key has a default.
+        # But don't skip if the instance was initialized with an explicit pk value
+        pk_field = self._meta.pk
+        explicitly_specified_pk = pk_field.attname in self.__dict__
         if (
             not force_insert and
             self._state.adding and
-            self._meta.pk.default and
-            self._meta.pk.default is not NOT_PROVIDED
+            pk_field.default and
+            pk_field.default is not NOT_PROVIDED and
+            not explicitly_specified_pk
         ):
             force_insert = True
         # If possible, try an UPDATE. If that doesn't update anything, do an INSERT.

