diff --git a/sympy/geometry/point.py b/sympy/geometry/point.py
index 73c20334d0..ad06668efe 100644
--- a/sympy/geometry/point.py
+++ b/sympy/geometry/point.py
@@ -152,7 +152,7 @@ def __new__(cls, *args, **kwargs):
                         'warn' or 'ignore'.'''))
         if any(coords[dim:]):
             raise ValueError('Nonzero coordinates cannot be removed.')
-        if any(a.is_number and im(a) for a in coords):
+        if evaluate and any(a.is_number and im(a) for a in coords):
             raise ValueError('Imaginary coordinates are not permitted.')
         if not all(isinstance(a, Expr) for a in coords):
             raise TypeError('Coordinates must be valid SymPy expressions.')
diff --git a/sympy/parsing/sympy_parser.py b/sympy/parsing/sympy_parser.py
index e403654cb4..885868573b 100644
--- a/sympy/parsing/sympy_parser.py
+++ b/sympy/parsing/sympy_parser.py
@@ -1130,6 +1130,7 @@ class EvaluateFalseTransformer(ast.NodeTransformer):
         'cos', 'cot', 'csc', 'sec', 'sin', 'tan',
         'cosh', 'coth', 'csch', 'sech', 'sinh', 'tanh',
         'exp', 'ln', 'log', 'sqrt', 'cbrt',
+        'Point2D', 'Point3D', 'Point',
     )
 
     def flatten(self, args, func):

