diff --git a/requests/models.py b/requests/models.py
index f82f56a3..2d49061f 100644
--- a/requests/models.py
+++ b/requests/models.py
@@ -307,7 +307,7 @@ class PreparedRequest(RequestEncodingMixin, RequestHooksMixin):
         """Prepares the given HTTP method."""
         self.method = method
         if self.method is not None:
-            self.method = self.method.upper()
+            self.method = to_native_string(self.method.upper())
 
     def prepare_url(self, url, params):
         """Prepares the given HTTP URL."""
diff --git a/requests/sessions.py b/requests/sessions.py
index cc72f65d..20dc42f7 100644
--- a/requests/sessions.py
+++ b/requests/sessions.py
@@ -310,8 +310,10 @@ class Session(SessionRedirectMixin):
             If Tuple, ('cert', 'key') pair.
         """
         # Create the Request.
+        # Convert the method to a native string to avoid Unicode issues
+        from .utils import to_native_string
         req = Request(
-            method = method.upper(),
+            method = to_native_string(method.upper()),
             url = url,
             headers = headers,
             files = files,

