diff --git a/django/db/migrations/operations/models.py b/django/db/migrations/operations/models.py
index dddf7c679c..7c1aeb4c6c 100644
--- a/django/db/migrations/operations/models.py
+++ b/django/db/migrations/operations/models.py
@@ -137,11 +137,19 @@ class CreateModel(ModelOperation):
                 ),
             ]
         elif isinstance(operation, AlterModelOptions) and self.name_lower == operation.name_lower:
+            # First merge the options
+            new_options = {**self.options, **operation.options}
+            
+            # Then remove keys that aren't in operation.options but are in ALTER_OPTION_KEYS
+            for key in operation.ALTER_OPTION_KEYS:
+                if key not in operation.options and key in new_options:
+                    new_options.pop(key)
+                    
             return [
                 CreateModel(
                     self.name,
                     fields=self.fields,
-                    options={**self.options, **operation.options},
+                    options=new_options,
                     bases=self.bases,
                     managers=self.managers,
                 ),

