diff --git a/django/forms/models.py b/django/forms/models.py
index 3fa04b821f..00be22c317 100644
--- a/django/forms/models.py
+++ b/django/forms/models.py
@@ -1133,7 +1133,16 @@ class BaseInlineFormSet(BaseModelFormSet):
         # Set the fk value here so that the form can do its validation.
         fk_value = self.instance.pk
         if self.fk.remote_field.field_name != self.fk.remote_field.model._meta.pk.name:
-            fk_value = getattr(self.instance, self.fk.remote_field.field_name)
+            remote_field = self.fk.remote_field.field_name
+            field = self.instance._meta.get_field(remote_field)
+            
+            # Check if the instance is new and the field has a default but hasn't been set
+            if self.instance._state.adding and field.has_default() and getattr(self.instance, remote_field) is None:
+                # Apply the default
+                default = field.get_default()
+                setattr(self.instance, remote_field, default() if callable(default) else default)
+            
+            fk_value = getattr(self.instance, remote_field)
             fk_value = getattr(fk_value, "pk", fk_value)
         setattr(form.instance, self.fk.get_attname(), fk_value)
         return form
