diff --git a/django/db/migrations/loader.py b/django/db/migrations/loader.py
index 95a5062ec9..99cc06fd05 100644
--- a/django/db/migrations/loader.py
+++ b/django/db/migrations/loader.py
@@ -15,6 +15,25 @@ from .exceptions import (
 MIGRATIONS_MODULE_NAME = 'migrations'
 
 
+def _is_namespace_package(module):
+    """
+    Identify if a module is a namespace package per PEP 420.
+    
+    A namespace package has:
+    - a __path__ attribute
+    - module.__spec__.origin == 'namespace' (preferred, PEP-451)
+    - or, as a fallback, its __path__ isn't a normal list
+    """
+    return (
+        hasattr(module, '__path__')
+        and (
+            (getattr(module, '__spec__', None) and
+             getattr(module.__spec__, 'origin', None) == 'namespace')
+            or not isinstance(module.__path__, list)
+        )
+    )
+
+
 class MigrationLoader:
     """
     Load migration files from disk and their status from the database.
@@ -89,8 +108,7 @@ class MigrationLoader:
                 raise
             else:
                 # Empty directories are namespaces.
-                # getattr() needed on PY36 and older (replace w/attribute access).
-                if getattr(module, '__file__', None) is None:
+                if _is_namespace_package(module):
                     self.unmigrated_apps.add(app_config.label)
                     continue
                 # Module is not a package (e.g. migrations.py).
