diff --git a/src/_pytest/unittest.py b/src/_pytest/unittest.py
index 851e4943b..94bd06236 100644
--- a/src/_pytest/unittest.py
+++ b/src/_pytest/unittest.py
@@ -204,8 +204,15 @@ class TestCaseFunction(Function):
             self._request._fillfixtures()
 
     def teardown(self) -> None:
+        is_skipped = False
+        if hasattr(self, "_excinfo") and self._excinfo:
+            is_skipped = any(
+                isinstance(exc.value, skip.Exception)
+                for exc in self._excinfo
+            )
         if self._explicit_tearDown is not None:
-            self._explicit_tearDown()
+            if not is_skipped:
+                self._explicit_tearDown()
             self._explicit_tearDown = None
         self._testcase = None
         self._obj = None
@@ -316,7 +323,11 @@ class TestCaseFunction(Function):
             # Arguably we could always postpone tearDown(), but this changes the moment where the
             # TestCase instance interacts with the results object, so better to only do it
             # when absolutely needed.
-            if self.config.getoption("usepdb") and not _is_skipped(self.obj):
+            if (
+                self.config.getoption("usepdb")
+                and not _is_skipped(self.obj)
+                and not _is_skipped(self._testcase)
+            ):
                 self._explicit_tearDown = self._testcase.tearDown
                 setattr(self._testcase, "tearDown", lambda *args: None)
 
