diff --git a/sklearn/utils/_show_versions.py b/sklearn/utils/_show_versions.py
index e5d3141bf..ecba9ec68 100644
--- a/sklearn/utils/_show_versions.py
+++ b/sklearn/utils/_show_versions.py
@@ -48,6 +48,7 @@ def _get_deps_info():
         "Cython",
         "pandas",
         "matplotlib",
+        "joblib",
     ]
 
     def get_version(module):
diff --git a/sklearn/utils/tests/test_show_versions.py b/sklearn/utils/tests/test_show_versions.py
index 31153c96c..f09393adc 100644
--- a/sklearn/utils/tests/test_show_versions.py
+++ b/sklearn/utils/tests/test_show_versions.py
@@ -23,6 +23,7 @@ def test_get_deps_info():
     assert 'Cython' in deps_info
     assert 'pandas' in deps_info
     assert 'matplotlib' in deps_info
+    assert 'joblib' in deps_info
 
 
 def test_show_versions_with_blas(capsys):
@@ -31,3 +32,4 @@ def test_show_versions_with_blas(capsys):
     assert 'python' in out
     assert 'numpy' in out
     assert 'BLAS' in out
+    assert 'joblib' in out
