diff --git a/django/db/models/expressions.py b/django/db/models/expressions.py
index 81f8f79c71..3a6edf3212 100644
--- a/django/db/models/expressions.py
+++ b/django/db/models/expressions.py
@@ -1209,20 +1209,24 @@ class Exists(Subquery):
         clone.negated = not self.negated
         return clone
 
     def as_sql(self, compiler, connection, template=None, **extra_context):
         query = self.query.exists(using=connection.alias)
-        sql, params = super().as_sql(
-            compiler,
-            connection,
-            template=template,
-            query=query,
-            **extra_context,
-        )
-        if self.negated:
-            sql = 'NOT {}'.format(sql)
-        return sql, params
+        try:
+            sql, params = super().as_sql(
+                compiler,
+                connection,
+                template=template,
+                query=query,
+                **extra_context,
+            )
+            if self.negated:
+                sql = 'NOT {}'.format(sql)
+            return sql, params
+        except EmptyResultSet:
+            # For an empty queryset, EXISTS returns FALSE, and NOT EXISTS returns TRUE
+            return ('0 = 1' if not self.negated else '1 = 1'), []
 
     def select_format(self, compiler, sql, params):
         # Wrap EXISTS() with a CASE WHEN expression if a database backend
         # (e.g. Oracle) doesn't support boolean expression in SELECT or GROUP
         # BY list.
