diff --git a/sympy/core/basic.py b/sympy/core/basic.py
index db4e971848..6ab9ce5273 100644
--- a/sympy/core/basic.py
+++ b/sympy/core/basic.py
@@ -311,11 +311,11 @@ def __eq__(self, other):
             if isinstance(other, Pow) and other.exp == 1:
                 return self == other.base
             try:
                 other = _sympify(other)
             except SympifyError:
-                return False    # sympy != other
+                return NotImplemented    # sympy != other
 
             if type(self) != type(other):
                 return False
 
         return self._hashable_content() == other._hashable_content()
diff --git a/sympy/core/numbers.py b/sympy/core/numbers.py
index 864d055087..6ede21177d 100644
--- a/sympy/core/numbers.py
+++ b/sympy/core/numbers.py
@@ -1256,11 +1256,11 @@ def __eq__(self, other):
                 return False
             return bool(mlib.mpf_eq(self._mpf_, ompf))
         try:
             other = _sympify(other)
         except SympifyError:
-            return False    # sympy != other  -->  not ==
+            return NotImplemented    # sympy != other  -->  not ==
         if isinstance(other, NumberSymbol):
             if other.is_irrational:
                 return False
             return other.__eq__(self)
         if isinstance(other, Float):
@@ -1717,11 +1717,11 @@ def ceiling(self):
 
     def __eq__(self, other):
         try:
             other = _sympify(other)
         except SympifyError:
-            return False    # sympy != other  -->  not ==
+            return NotImplemented    # sympy != other  -->  not ==
         if isinstance(other, NumberSymbol):
             if other.is_irrational:
                 return False
             return other.__eq__(self)
         if isinstance(other, Number):
@@ -3337,11 +3337,11 @@ def _eval_evalf(self, prec):
 
     def __eq__(self, other):
         try:
             other = _sympify(other)
         except SympifyError:
-            return False    # sympy != other  -->  not ==
+            return NotImplemented    # sympy != other  -->  not ==
         if self is other:
             return True
         if isinstance(other, Number) and self.is_irrational:
             return False
 
