diff --git a/django/db/migrations/operations/fields.py b/django/db/migrations/operations/fields.py
index cd3aab43ad..e2682aeac0 100644
--- a/django/db/migrations/operations/fields.py
+++ b/django/db/migrations/operations/fields.py
@@ -262,10 +262,14 @@ class AlterField(FieldOperation):
                     model_name=self.model_name,
                     name=operation.new_name,
                     field=self.field,
                 ),
             ]
+        elif isinstance(operation, AlterField) and self.is_same_field_operation(
+            operation
+        ):
+            return [operation]
         return super().reduce(operation, app_label)
 
 
 class RenameField(FieldOperation):
     """Rename a field on the model. Might affect db_column too."""
