diff --git a/lib/matplotlib/dates.py b/lib/matplotlib/dates.py
index 0766156d64..193098115c 100644
--- a/lib/matplotlib/dates.py
+++ b/lib/matplotlib/dates.py
@@ -590,17 +590,14 @@ def drange(dstart, dend, delta):
     f2 = date2num(dinterval_end)  # new float-endpoint
     return np.linspace(f1, f2, num + 1)
 
 
 def _wrap_in_tex(text):
-    p = r'([a-zA-Z]+)'
-    ret_text = re.sub(p, r'}$\1$\\mathdefault{', text)
-
     # Braces ensure dashes are not spaced like binary operators.
-    ret_text = '$\\mathdefault{'+ret_text.replace('-', '{-}')+'}$'
-    ret_text = ret_text.replace('$\\mathdefault{}$', '')
-    return ret_text
+    # Reverting to the simpler implementation from matplotlib 3.3
+    # to fix spacing issues with datetime axes when usetex is enabled
+    return '$\\mathdefault{' + text.replace('-', '{-}') + '}$'
 
 
 ## date tickers and formatters ###
 
 
