diff --git a/django/db/models/lookups.py b/django/db/models/lookups.py
index 866e38df83..b5bbb6d354 100644
--- a/django/db/models/lookups.py
+++ b/django/db/models/lookups.py
@@ -418,13 +418,12 @@ class In(FieldGetDbPrepValueIterableMixin, BuiltinLookup):
     def get_prep_lookup(self):
         from django.db.models.sql.query import Query  # avoid circular import
 
         if isinstance(self.rhs, Query):
             self.rhs.clear_ordering(clear_default=True)
-            if not self.rhs.has_select_fields:
-                self.rhs.clear_select_clause()
-                self.rhs.add_fields(["pk"])
+            self.rhs.clear_select_clause()
+            self.rhs.add_fields(["pk"])
         return super().get_prep_lookup()
 
     def process_rhs(self, compiler, connection):
         db_rhs = getattr(self.rhs, "_db", None)
         if db_rhs is not None and db_rhs != connection.alias:
diff --git a/tests/annotations/tests.py b/tests/annotations/tests.py
index 8de2bf1998..6f2dd4abc8 100644
--- a/tests/annotations/tests.py
+++ b/tests/annotations/tests.py
@@ -1107,10 +1107,34 @@ class NonAggregateAnnotationTestCase(TestCase):
         )
         for crafted_alias in tests:
             with self.subTest(crafted_alias):
                 with self.assertRaisesMessage(ValueError, msg):
                     Book.objects.annotate(**{crafted_alias: Value(1)})
+    
+    def test_annotation_and_alias_filter_in_subquery(self):
+        long_books_qs = (
+            Book.objects.filter(
+                pages__gt=400,
+            )
+            .annotate(book_annotate=Value(1))
+            .alias(book_alias=Value(1))
+        )
+        publisher_books_qs = (
+            Publisher.objects.filter(
+                book__in=long_books_qs
+            )
+            .values("name")
+        )
+        self.assertCountEqual(
+            publisher_books_qs,
+            [
+                {'name': 'Apress'},
+                {'name': 'Sams'},
+                {'name': 'Prentice Hall'},
+                {'name': 'Morgan Kaufmann'}
+            ]
+        )
 
 
 class AliasTests(TestCase):
     @classmethod
     def setUpTestData(cls):
