diff --git a/django/db/migrations/writer.py b/django/db/migrations/writer.py
index a59f0c8dcb..ef1790f7a5 100644
--- a/django/db/migrations/writer.py
+++ b/django/db/migrations/writer.py
@@ -171,13 +171,16 @@ class MigrationWriter:
             imports.discard("from django.db import models")
             imports.add("from django.db import migrations, models")
         else:
             imports.add("from django.db import migrations")
 
-        # Sort imports by the package / module to be imported (the part after
-        # "from" in "from ... import ..." or after "import" in "import ...").
-        sorted_imports = sorted(imports, key=lambda i: i.split()[1])
+        # Sort imports according to Django coding style:
+        # 1. All import module statements
+        # 2. All from module import objects statements
+        import_statements = sorted([i for i in imports if i.startswith('import')])
+        from_statements = sorted([i for i in imports if i.startswith('from')])
+        sorted_imports = import_statements + from_statements
         items["imports"] = "\n".join(sorted_imports) + "\n" if imports else ""
         if migration_imports:
             items["imports"] += (
                 "\n\n# Functions from the following migrations need manual "
                 "copying.\n# Move them and any dependencies into this file, "
