diff --git a/pylint/config/utils.py b/pylint/config/utils.py
index 8c790416f..cdbde612c 100644
--- a/pylint/config/utils.py
+++ b/pylint/config/utils.py
@@ -205,10 +205,11 @@ PREPROCESSABLE_OPTIONS: dict[
     "--init-hook": (True, _init_hook),
     "--rcfile": (True, _set_rcfile),
     "--output": (True, _set_output),
     "--load-plugins": (True, _add_plugins),
     "--verbose": (False, _set_verbose_mode),
+    "-v": (False, _set_verbose_mode),  # Add the short option
     "--enable-all-extensions": (False, _enable_all_extensions),
 }
 
 
 def _preprocess_options(run: Run, args: Sequence[str]) -> list[str]:
@@ -216,11 +217,11 @@ def _preprocess_options(run: Run, args: Sequence[str]) -> list[str]:
     processed_args: list[str] = []
 
     i = 0
     while i < len(args):
         argument = args[i]
-        if not argument.startswith("--"):
+        if not (argument.startswith("--") or argument.startswith("-")):
             processed_args.append(argument)
             i += 1
             continue
 
         try:
@@ -239,11 +240,11 @@ def _preprocess_options(run: Run, args: Sequence[str]) -> list[str]:
             i += 1
             if i >= len(args) or args[i].startswith("-"):
                 raise ArgumentPreprocessingError(f"Option {option} expects a value")
             value = args[i]
         elif not takearg and value is not None:
-            raise ArgumentPreprocessingError(f"Option {option} doesn't expects a value")
+            raise ArgumentPreprocessingError(f"Option {option} doesn't expect a value")
 
         cb(run, value)
         i += 1
 
     return processed_args
