diff --git b/sphinx/util/typing.py a/sphinx/util/typing.py
index 8d5518da9..87707d48f 100644
--- b/sphinx/util/typing.py
+++ a/sphinx/util/typing.py
@@ -115,13 +115,10 @@ def restify(cls: Optional[Type]) -> str:
             return '...'
         elif isinstance(cls, str):
             return cls
         elif cls in INVALID_BUILTIN_CLASSES:
             return ':py:class:`%s`' % INVALID_BUILTIN_CLASSES[cls]
-        # Check for __display_name__ attribute which is used by mocked objects
-        elif hasattr(cls, '__display_name__'):
-            return ':py:class:`%s`' % cls.__display_name__
         elif inspect.isNewType(cls):
             if sys.version_info > (3, 10):
                 # newtypes have correct module info since Python 3.10+
                 print(cls, type(cls), dir(cls))
                 return ':py:class:`%s.%s`' % (cls.__module__, cls.__name__)
