diff --git a/sympy/matrices/matrices.py b/sympy/matrices/matrices.py
index 97cde62e26..ec7678e4a8 100644
--- a/sympy/matrices/matrices.py
+++ b/sympy/matrices/matrices.py
@@ -421,11 +421,15 @@ def det(self, method="bareiss"):
                    - self[0, 2] * self[1, 1] * self[2, 0]
                    - self[0, 0] * self[1, 2] * self[2, 1]
                    - self[0, 1] * self[1, 0] * self[2, 2])
 
         if method == "bareiss":
-            return self._eval_det_bareiss()
+            if self.free_symbols:
+                # Use Berkowitz algorithm for symbolic matrices
+                return self._eval_det_berkowitz()
+            else:
+                return self._eval_det_bareiss()
         elif method == "berkowitz":
             return self._eval_det_berkowitz()
         elif method == "lu":
             return self._eval_det_lu()
 
