diff --git a/sympy/ntheory/residue_ntheory.py b/sympy/ntheory/residue_ntheory.py
index 9a3052d4c3..4c1fa9bf6e 100644
--- a/sympy/ntheory/residue_ntheory.py
+++ b/sympy/ntheory/residue_ntheory.py
@@ -771,10 +771,18 @@ def nthroot_mod(a, n, p, all_roots=False):
     if n == 2:
         return sqrt_mod(a, p, all_roots)
     # see Hackman "Elementary Number Theory" (2009), page 76
     if not is_nthpow_residue(a, n, p):
         return None
+    
+    # Special case: if a % p == 0, then x = 0 is a root
+    if a % p == 0:
+        if all_roots:
+            return [0]
+        else:
+            return 0
+            
     if not isprime(p):
         raise NotImplementedError("Not implemented for composite p")
 
     if (p - 1) % n == 0:
         return _nthroot_mod1(a, n, p, all_roots)
