diff --git b/sphinx/domains/python.py a/sphinx/domains/python.py
index def73145e..3ca730e30 100644
--- b/sphinx/domains/python.py
+++ a/sphinx/domains/python.py
@@ -98,12 +98,11 @@ def _parse_annotation(annotation: str) -> List[Node]:
         elif isinstance(node, ast.List):
             result = [addnodes.desc_sig_punctuation('', '[')]  # type: List[Node]
             for elem in node.elts:
                 result.extend(unparse(elem))
                 result.append(addnodes.desc_sig_punctuation('', ', '))
-            if node.elts:
-                result.pop()
+            result.pop()
             result.append(addnodes.desc_sig_punctuation('', ']'))
             return result
         elif isinstance(node, ast.Module):
             return sum((unparse(e) for e in node.body), [])
         elif isinstance(node, ast.Name):
@@ -117,12 +116,11 @@ def _parse_annotation(annotation: str) -> List[Node]:
         elif isinstance(node, ast.Tuple):
             result = []
             for elem in node.elts:
                 result.extend(unparse(elem))
                 result.append(addnodes.desc_sig_punctuation('', ', '))
-            if node.elts:
-                result.pop()
+            result.pop()
             return result
         else:
             raise SyntaxError  # unsupported syntax
 
     try:
