diff --git a/lib/matplotlib/__init__.py b/lib/matplotlib/__init__.py
index c268a56724..17a6dbaf9e 100644
--- a/lib/matplotlib/__init__.py
+++ b/lib/matplotlib/__init__.py
@@ -1093,11 +1093,11 @@ def rc_context(rc=None, fname=None):
             rc_file(fname)
         if rc:
             rcParams.update(rc)
         yield
     finally:
-        dict.update(rcParams, orig)  # Revert to the original rcs.
+        rcParams.update(orig)  # Revert to the original rcs.
 
 
 def use(backend, *, force=True):
     """
     Select the backend used for rendering and GUI integration.
