diff --git a/django/db/models/fields/__init__.py b/django/db/models/fields/__init__.py
index 0f53d9c30b..900f92417f 100644
--- a/django/db/models/fields/__init__.py
+++ b/django/db/models/fields/__init__.py
@@ -2522,11 +2522,15 @@ class AutoFieldMeta(type):
 
     def __instancecheck__(self, instance):
         return isinstance(instance, self._subclasses) or super().__instancecheck__(instance)
 
     def __subclasscheck__(self, subclass):
-        return subclass in self._subclasses or super().__subclasscheck__(subclass)
+        try:
+            return any(issubclass(subclass, cls) for cls in self._subclasses) or super().__subclasscheck__(subclass)
+        except TypeError:
+            # subclass is not a class
+            return super().__subclasscheck__(subclass)
 
 
 class AutoField(AutoFieldMixin, IntegerField, metaclass=AutoFieldMeta):
 
     def get_internal_type(self):
