diff --git b/sphinx/cmd/quickstart.py a/sphinx/cmd/quickstart.py
index bd714602c..a40a21073 100644
--- b/sphinx/cmd/quickstart.py
+++ a/sphinx/cmd/quickstart.py
@@ -221,19 +221,13 @@ def ask_user(d: Dict) -> None:
         print(bold(__('Error: an existing conf.py has been found in the '
                       'selected root path.')))
         print(__('sphinx-quickstart will not overwrite existing Sphinx projects.'))
         print()
         d['path'] = do_prompt(__('Please enter a new root path (or just Enter to exit)'),
-                              '', allow_empty)
+                              '', is_path)
         if not d['path']:
             sys.exit(1)
-        # Validate the path if it's not empty
-        try:
-            d['path'] = is_path(d['path'])
-        except ValidationError as err:
-            print(red('* ' + str(err)))
-            continue
 
     if 'sep' not in d:
         print()
         print(__('You have two options for placing the build directory for Sphinx output.\n'
                  'Either, you use a directory "_build" within the root path, or you separate\n'
