diff --git b/sphinx/domains/python.py a/sphinx/domains/python.py
index a53639a6e..79d7e4f46 100644
--- b/sphinx/domains/python.py
+++ a/sphinx/domains/python.py
@@ -1363,16 +1363,10 @@ class PythonDomain(Domain):
         if target is None:
             return None
         else:
             return '.'.join(filter(None, [modname, clsname, target]))
 
-    def process_field_xref(self, pnode: pending_xref) -> None:
-        """Process a pending xref created in a doc field.
-        Attach information about the current scope.
-        """
-        pnode.attributes.update(self.env.ref_context)
-
 
 def builtin_resolver(app: Sphinx, env: BuildEnvironment,
                      node: pending_xref, contnode: Element) -> Element:
     """Do not emit nitpicky warnings for built-in types."""
     def istyping(s: str) -> bool:
