diff --git a/sympy/core/assumptions.py b/sympy/core/assumptions.py
index ce4dd30f2e..a198800f31 100644
--- a/sympy/core/assumptions.py
+++ b/sympy/core/assumptions.py
@@ -184,10 +184,12 @@
 
     'nonpositive    ==  real & !positive',
     'nonnegative    ==  real & !negative',
 
     'zero           ->  even & finite',
+    'integer        ->  finite',
+    'rational       ->  finite',
 
     'prime          ->  integer & positive',
     'composite      ->  integer & positive & !prime',
     '!composite     ->  !positive | !even | prime',
 
