diff --git a/django/core/servers/basehttp.py b/django/core/servers/basehttp.py
index 14dab517c9..527f7b5976 100644
--- a/django/core/servers/basehttp.py
+++ b/django/core/servers/basehttp.py
@@ -78,10 +78,22 @@ class WSGIServer(simple_server.WSGIServer):
 
 
 class ThreadedWSGIServer(socketserver.ThreadingMixIn, WSGIServer):
     """A threaded version of the WSGIServer"""
     daemon_threads = True
+    
+    def process_request_thread(self, request, client_address):
+        """Same as in ThreadingMixIn but closes database connections."""
+        try:
+            self.finish_request(request, client_address)
+            self.shutdown_request(request)
+        except Exception:
+            self.handle_error(request, client_address)
+            self.shutdown_request(request)
+        finally:
+            from django.db import connections
+            connections.close_all()
 
 
 class ServerHandler(simple_server.ServerHandler):
     http_version = '1.1'
 
