diff --git b/sphinx/ext/autodoc/__init__.py a/sphinx/ext/autodoc/__init__.py
index bf1d3cbd7..76265f5be 100644
--- b/sphinx/ext/autodoc/__init__.py
+++ a/sphinx/ext/autodoc/__init__.py
@@ -727,22 +727,10 @@ class Documenter:
                 doc = obj.docstring
 
             has_doc = bool(doc)
 
             metadata = extract_metadata(doc)
-            
-            # Check for comment docstrings for variables
-            if self.analyzer and membername.startswith('_'):
-                namespace = '.'.join(self.objpath)  # will be empty for modules
-                attr_docs = self.analyzer.find_attr_docs()
-                if (namespace, membername) in attr_docs:
-                    # Join the comment docstring lines into a single string for metadata extraction
-                    comment_doc = '\n'.join(attr_docs[(namespace, membername)])
-                    comment_metadata = extract_metadata(comment_doc)
-                    # Update metadata with any found in the comment docstring
-                    metadata.update(comment_metadata)
-            
             if 'private' in metadata:
                 # consider a member private if docstring has "private" metadata
                 isprivate = True
             elif 'public' in metadata:
                 # consider a member public if docstring has "public" metadata
