--- a/django/contrib/auth/forms.py
+++ b/django/contrib/auth/forms.py
@@ -56,15 +56,8 @@
 
     def __init__(self, *args, **kwargs):
         kwargs.setdefault("required", False)
-        super().__init__(*args, **kwargs)
-
-    def bound_data(self, data, initial):
-        # Always return initial because the widget doesn't
-        # render an input field.
-        return initial
-
-    def has_changed(self, initial, data):
-        return False
+        kwargs.setdefault("disabled", True)
+        super().__init__(*args, **kwargs)
 
 
 class UsernameField(forms.CharField):
@@ -162,12 +155,6 @@
         user_permissions = self.fields.get('user_permissions')
         if user_permissions:
             user_permissions.queryset = user_permissions.queryset.select_related('content_type')
-
-    def clean_password(self):
-        # Regardless of what the user provides, return the initial value.
-        # This is done here, rather than on the field, because the
-        # field does not have access to the initial value
-        return self.initial.get('password')
 
 
 class AuthenticationForm(forms.Form):
