--- a/lib/matplotlib/axes/_axes.py
+++ b/lib/matplotlib/axes/_axes.py
@@ -2180,12 +2180,12 @@
             # wrap numpy arrays.
             try:
                 x0 = cbook._safe_first_finite(x0)
-            except (TypeError, IndexError, KeyError):
+            except (TypeError, IndexError, KeyError, StopIteration):
                 pass
 
             try:
                 x = cbook._safe_first_finite(xconv)
-            except (TypeError, IndexError, KeyError):
+            except (TypeError, IndexError, KeyError, StopIteration):
                 x = xconv
 
             delist = False
@@ -2829,11 +2829,17 @@
         """
         # process the unit information
         if len(xranges):
-            xdata = cbook._safe_first_finite(xranges)
+            try:
+                xdata = cbook._safe_first_finite(xranges)
+            except (TypeError, IndexError, KeyError, StopIteration):
+                xdata = None
         else:
             xdata = None
         if len(yrange):
-            ydata = cbook._safe_first_finite(yrange)
+            try:
+                ydata = cbook._safe_first_finite(yrange)
+            except (TypeError, IndexError, KeyError, StopIteration):
+                ydata = None
         else:
             ydata = None
         self._process_unit_info(
