--- a/sklearn/linear_model/logistic.py
+++ b/sklearn/linear_model/logistic.py
@@ -922,7 +922,12 @@
         check_input=False, max_squared_sum=max_squared_sum,
         sample_weight=sample_weight)
 
-    log_reg = LogisticRegression(fit_intercept=fit_intercept)
+    log_reg = LogisticRegression(
+        fit_intercept=fit_intercept, multi_class=multi_class,
+        solver=solver, penalty=penalty, dual=dual,
+        class_weight=class_weight, max_iter=max_iter, tol=tol,
+        intercept_scaling=intercept_scaling, random_state=random_state,
+        verbose=verbose)
 
     # The score method of Logistic Regression has a classes_ attribute.
     if multi_class == 'ovr':
