diff --git a/sphinx/domains/python.py b/sphinx/domains/python.py
index 7d39d80ed..8f9c8ae46 100644
--- a/sphinx/domains/python.py
+++ b/sphinx/domains/python.py
@@ -288,8 +288,9 @@ class PyXrefMixin:
         result = super().make_xref(rolename, domain, target,  # type: ignore
                                    innernode, contnode, env)
         result['refspecific'] = True
-        result['py:module'] = env.ref_context.get('py:module')
-        result['py:class'] = env.ref_context.get('py:class')
+        if env is not None:  # guard for out-of-band callers
+            result['py:module'] = env.ref_context.get('py:module')
+            result['py:class'] = env.ref_context.get('py:class')
         if target.startswith(('.', '~')):
             prefix, result['reftarget'] = target[0], target[1:]
             if prefix == '.':
@@ -304,7 +305,8 @@ class PyXrefMixin:
     def make_xrefs(self, rolename: str, domain: str, target: str,
                    innernode: Type[TextlikeNode] = nodes.emphasis,
                    contnode: Node = None, env: BuildEnvironment = None) -> List[Node]:
-        delims = r'(\s*[\[\]\(\),](?:\s*or\s)?\s*|\s+or\s+|\.\.\.)'
+        # also split on "|" (PEP-604 union operator)
+        delims = r'(\s*[\[\]\(\),|](?:\s*or\s)?\s*|\s+or\s+|\.\.\.)'
         delims_re = re.compile(delims)
         sub_targets = re.split(delims, target)
 
