diff --git a/sklearn/model_selection/_split.py b/sklearn/model_selection/_split.py
index c49a3ce6a..b84d42e56 100644
--- a/sklearn/model_selection/_split.py
+++ b/sklearn/model_selection/_split.py
@@ -1087,6 +1087,9 @@ class _RepeatedSplits(metaclass=ABCMeta):
         Constructor parameters for cv. Must not contain random_state
         and shuffle.
     """
+    
+    def __repr__(self):
+        return _build_repr(self)
     def __init__(self, cv, n_repeats=10, random_state=None, **cvargs):
         if not isinstance(n_repeats, numbers.Integral):
             raise ValueError("Number of repetitions must be of Integral type.")
@@ -1215,6 +1218,10 @@ class RepeatedKFold(_RepeatedSplits):
     def __init__(self, n_splits=5, n_repeats=10, random_state=None):
         super().__init__(
             KFold, n_repeats, random_state, n_splits=n_splits)
+            
+    @property
+    def n_splits(self):
+        return self.cvargs.get('n_splits', 5)
 
 
 class RepeatedStratifiedKFold(_RepeatedSplits):
@@ -1268,6 +1275,10 @@ class RepeatedStratifiedKFold(_RepeatedSplits):
     def __init__(self, n_splits=5, n_repeats=10, random_state=None):
         super().__init__(
             StratifiedKFold, n_repeats, random_state, n_splits=n_splits)
+            
+    @property
+    def n_splits(self):
+        return self.cvargs.get('n_splits', 5)
 
 
 class BaseShuffleSplit(metaclass=ABCMeta):
