diff --git a/django/db/models/base.py b/django/db/models/base.py
index 3ce711f3f1..7c79d2bed5 100644
--- a/django/db/models/base.py
+++ b/django/db/models/base.py
@@ -1707,6 +1707,13 @@ class Model(metaclass=ModelBase):
             _cls = cls
             fld = None
             for part in field.split(LOOKUP_SEP):
+                # Special-case the primary-key alias "pk".
+                if part == 'pk':
+                    fld = _cls._meta.pk
+                    if fld.is_relation:
+                        _cls = fld.get_path_info()[-1].to_opts.model
+                    # 'pk' is always valid, move on to the next path component.
+                    continue
                 try:
                     fld = _cls._meta.get_field(part)
                     if fld.is_relation:
