diff --git a/sympy/utilities/lambdify.py b/sympy/utilities/lambdify.py
index da67263b19..5c5473e639 100644
--- a/sympy/utilities/lambdify.py
+++ b/sympy/utilities/lambdify.py
@@ -958,6 +958,9 @@ def _recursive_to_string(doprint, arg):
         if isinstance(arg, list):
             left, right = "[]"
         elif isinstance(arg, tuple):
+            if len(arg) == 1:
+                # Special case for single-element tuples: need trailing comma
+                return '(' + _recursive_to_string(doprint, arg[0]) + ',)'
             left, right = "()"
         else:
             raise NotImplementedError("unhandled type: %s, %s" % (type(arg), arg))
