diff --git a/xarray/core/dataarray.py b/xarray/core/dataarray.py
index b0df8749..d346f5d1 100644
--- a/xarray/core/dataarray.py
+++ b/xarray/core/dataarray.py
@@ -1958,11 +1958,15 @@ class DataArray(AbstractArray, DataWithCoords):
         variables = idx.levels[level_number]
         variable_dim = idx.names[level_number]
 
-        # pull variables out of datarray
+        # pull variables out of DataArray, dropping the now-scalar stacked coord
         data_dict = {}
         for k in variables:
-            data_dict[k] = self.sel({variable_dim: k}).squeeze(drop=True)
-
+            da = self.sel({variable_dim: k}).squeeze(drop=True)
+            # drop the leftover 0-dim coord for the stacked dimension if it exists
+            if dim in da.coords:
+                da = da.drop_vars(dim)
+            data_dict[k] = da
+            
         # unstacked dataset
         return Dataset(data_dict)
 
