diff --git a/sympy/printing/pycode.py b/sympy/printing/pycode.py
index d06901425d..093d214a27 100644
--- a/sympy/printing/pycode.py
+++ b/sympy/printing/pycode.py
@@ -331,6 +331,11 @@ def _print_Float(self, e):
         args = str(tuple(map(int, e._mpf_)))
         return '{func}({args})'.format(func=self._module_format('mpmath.mpf'), args=args)
 
+    def _print_Rational(self, e):
+        # Convert Rational to mpmath mpf objects to maintain full precision
+        # during evaluation
+        mpf = self._module_format('mpmath.mpf')
+        return '{0}({1})/{0}({2})'.format(mpf, int(e.p), int(e.q))
 
     def _print_uppergamma(self, e):
         return "{0}({1}, {2}, {3})".format(
