diff --git a/sympy/printing/conventions.py b/sympy/printing/conventions.py
index 9a3be6bbe9..fb082d326c 100644
--- a/sympy/printing/conventions.py
+++ b/sympy/printing/conventions.py
@@ -7,7 +7,10 @@
 from collections.abc import Iterable
 from sympy import Derivative
 
-_name_with_digits_p = re.compile(r'^([a-zA-Z]+)([0-9]+)$')
+# Match "one-or-more Unicode letters" followed by "one-or-more digits".
+# [^\W\d_] == "word characters" minus "digits" and "_" => only letters.
+# The UNICODE flag is implicit on Py3 but added for clarity.
+_name_with_digits_p = re.compile(r'^([^\W\d_]+)(\d+)$', re.UNICODE)
 
 
 def split_super_sub(text):
