diff --git a/django/contrib/admin/helpers.py b/django/contrib/admin/helpers.py
index dd154f72ee..45d6e8e29d 100644
--- a/django/contrib/admin/helpers.py
+++ b/django/contrib/admin/helpers.py
@@ -204,7 +204,8 @@ class AdminReadonlyField:
         return format_html('<label{}>{}{}</label>', flatatt(attrs), capfirst(label), self.form.label_suffix)
 
     def get_admin_url(self, remote_field, remote_obj):
-        url_name = 'admin:%s_%s_change' % (
+        url_name = '%s:%s_%s_change' % (
+            self.model_admin.admin_site.name,
             remote_field.model._meta.app_label,
             remote_field.model._meta.model_name,
         )
