diff --git a/sympy/core/assumptions.py b/sympy/core/assumptions.py
index ce4dd30f2e..1149927aa0 100644
--- a/sympy/core/assumptions.py
+++ b/sympy/core/assumptions.py
@@ -187,6 +187,8 @@
 
     'zero           ->  even & finite',
 
+    'integer        ->  finite',
+    'rational       ->  finite',
     'prime          ->  integer & positive',
     'composite      ->  integer & positive & !prime',
     '!composite     ->  !positive | !even | prime',
