diff --git a/sympy/physics/vector/vector.py b/sympy/physics/vector/vector.py
index 77e79b6..ea4a1bf 100644
--- a/sympy/physics/vector/vector.py
+++ b/sympy/physics/vector/vector.py
@@ -56,10 +56,14 @@ class Vector(object):
         return hash(tuple(self.args))
 
     def __add__(self, other):
+
         """The add operator for Vector. """
+        if other == 0:
+            return self
         other = _check_vector(other)
         return Vector(self.args + other.args)
 
+
     def __and__(self, other):
         """Dot product of two vectors.
 
@@ -723,4 +727,4 @@ class VectorTypeError(TypeError):
 def _check_vector(other):
     if not isinstance(other, Vector):
         raise TypeError('A Vector must be supplied')
-    return other
+    return other
