diff --git a/django/db/models/aggregates.py b/django/db/models/aggregates.py
index 80dc7eb..d5f43a8 100644
--- a/django/db/models/aggregates.py
+++ b/django/db/models/aggregates.py
@@ -64,8 +64,12 @@ class Aggregate(Func):
             return c
         if hasattr(default, 'resolve_expression'):
             default = default.resolve_expression(query, allow_joins, reuse, summarize)
+
         c.default = None  # Reset the default argument before wrapping.
-        return Coalesce(c, default, output_field=c._output_field_or_none)
+        wrapper = Coalesce(c, default, output_field=c._output_field_or_none)
+        wrapper.is_summary = c.is_summary
+        return wrapper
+
 
     @property
     def default_alias(self):
@@ -162,4 +166,4 @@ class Variance(NumericOutputFieldMixin, Aggregate):
         super().__init__(expression, **extra)
 
     def _get_repr_options(self):
-        return {**super()._get_repr_options(), 'sample': self.function == 'VAR_SAMP'}
+        return {**super()._get_repr_options(), 'sample': self.function == 'VAR_SAMP'}
