diff --git a/django/db/backends/mysql/base.py b/django/db/backends/mysql/base.py
index 66049be..d28b735 100644
--- a/django/db/backends/mysql/base.py
+++ b/django/db/backends/mysql/base.py
@@ -199,10 +199,11 @@ class DatabaseWrapper(BaseDatabaseWrapper):
         settings_dict = self.settings_dict
         if settings_dict['USER']:
             kwargs['user'] = settings_dict['USER']
+
         if settings_dict['NAME']:
-            kwargs['db'] = settings_dict['NAME']
+            kwargs['database'] = settings_dict['NAME']
         if settings_dict['PASSWORD']:
-            kwargs['passwd'] = settings_dict['PASSWORD']
+            kwargs['password'] = settings_dict['PASSWORD']
         if settings_dict['HOST'].startswith('/'):
             kwargs['unix_socket'] = settings_dict['HOST']
         elif settings_dict['HOST']:
@@ -214,6 +215,11 @@ class DatabaseWrapper(BaseDatabaseWrapper):
         kwargs['client_flag'] = CLIENT.FOUND_ROWS
         # Validate the transaction isolation level, if specified.
         options = settings_dict['OPTIONS'].copy()
+        if 'db' in options and 'database' not in options:
+            options['database'] = options.pop('db')
+        if 'passwd' in options and 'password' not in options:
+            options['password'] = options.pop('passwd')
+
         isolation_level = options.pop('isolation_level', 'read committed')
         if isolation_level:
             isolation_level = isolation_level.lower()
