diff --git a/django/db/backends/base/schema.py b/django/db/backends/base/schema.py
index cfb9a2b..005cdac 100644
--- a/django/db/backends/base/schema.py
+++ b/django/db/backends/base/schema.py
@@ -1,6 +1,8 @@
 import logging
+import re
 from datetime import datetime
 
+
 from django.db.backends.ddl_references import (
     Columns, ForeignKeyName, IndexName, Statement, Table,
 )
@@ -1095,14 +1097,22 @@ class BaseDatabaseSchemaEditor:
             )
         return self._delete_constraint_sql(self.sql_delete_unique, model, name)
 
+
     def _check_sql(self, name, check):
+        if self.connection.vendor in ('sqlite', 'oracle'):
+            check = re.sub(r'"[^"]+"\.', '', check)
         return self.sql_constraint % {
+
             'name': self.quote_name(name),
             'constraint': self.sql_check_constraint % {'check': check},
         }
 
+
     def _create_check_sql(self, model, name, check):
+        if self.connection.vendor in ('sqlite', 'oracle'):
+            check = re.sub(r'"[^"]+"\.', '', check)
         return Statement(
+
             self.sql_create_check,
             table=Table(model._meta.db_table, self.quote_name),
             name=self.quote_name(name),
@@ -1177,4 +1187,4 @@ class BaseDatabaseSchemaEditor:
             'procedure': self.quote_name(procedure_name),
             'param_types': ','.join(param_types),
         }
-        self.execute(sql)
+        self.execute(sql)
