diff --git a/lib/matplotlib/contour.py b/lib/matplotlib/contour.py
index 3f84250b64..8a54387e7c 100644
--- a/lib/matplotlib/contour.py
+++ b/lib/matplotlib/contour.py
@@ -1459,6 +1459,18 @@ class ContourSet(ContourLabeler, mcoll.Collection):
 
         return (conmin, segmin, imin, xmin, ymin, d2min)
 
+    def set_paths(self, paths):
+        """
+        Set the paths of the contour lines/polygons.
+        
+        Parameters
+        ----------
+        paths : list of `~matplotlib.path.Path`
+            The paths to set
+        """
+        self._paths = paths
+        self.stale = True
+        
     def draw(self, renderer):
         paths = self._paths
         n_paths = len(paths)
