diff --git a/requests/sessions.py b/requests/sessions.py
index a023e4ec..09bb4140 100644
--- a/requests/sessions.py
+++ b/requests/sessions.py
@@ -55,9 +55,12 @@ def merge_setting(request_setting, session_setting, dict_class=OrderedDict):
     merged_setting.update(to_key_val_list(request_setting))
 
     # Remove keys that are set to None.
-    for (k, v) in request_setting.items():
+    for (k, v) in list(request_setting.items()):
         if v is None:
-            del merged_setting[k]
+            try:
+                del merged_setting[k]
+            except KeyError:
+                pass
 
     return merged_setting
 
diff --git a/requests/structures.py b/requests/structures.py
index a1759137..1d05a28a 100644
--- a/requests/structures.py
+++ b/requests/structures.py
@@ -71,7 +71,11 @@ class CaseInsensitiveDict(collections.MutableMapping):
     def __setitem__(self, key, value):
         # Use the lowercased key for lookups, but store the actual
         # key alongside the value.
-        self._store[key.lower()] = (key, value)
+        if value is None:
+            # If value is None, remove the key instead
+            self._store.pop(key.lower(), None)
+        else:
+            self._store[key.lower()] = (key, value)
 
     def __getitem__(self, key):
         return self._store[key.lower()][1]

