diff --git a/xarray/core/rolling.py b/xarray/core/rolling.py
index d923cdb..de840f2 100644
--- a/xarray/core/rolling.py
+++ b/xarray/core/rolling.py
@@ -401,8 +401,11 @@ class DataArrayRolling(Rolling["DataArray"]):
             attrs=attrs,
             name=obj.name,
         )
+
+        if hasattr(result, "_coord_names") and hasattr(self.obj, "_coord_names"): result._coord_names.update(self.obj._coord_names & result._variables.keys())
         return result.isel({d: slice(None, None, s) for d, s in zip(self.dim, strides)})
 
+
     def reduce(
         self, func: Callable, keep_attrs: bool | None = None, **kwargs: Any
     ) -> DataArray:
