diff --git a/astropy/io/ascii/html.py b/astropy/io/ascii/html.py
index 1220930..d07e4e6 100644
--- a/astropy/io/ascii/html.py
+++ b/astropy/io/ascii/html.py
@@ -346,10 +346,16 @@ class HTML(core.BaseReader):
         # Check that table has only 1-d or 2-d columns. Above that fails.
         self._check_multidim_table(table)
 
+
         cols = list(table.columns.values())
 
+        # Make formats supplied by ``formats=`` take effect
+        self.data.cols = cols
+        self.data._set_col_formats()
+
         self.data.header.cols = cols
 
+
         if isinstance(self.data.fill_values, tuple):
             self.data.fill_values = [self.data.fill_values]
 
@@ -428,9 +434,12 @@ class HTML(core.BaseReader):
                                 span = col.shape[1]
                                 for i in range(span):
                                     # Split up multicolumns into separate columns
+
                                     new_col = Column([el[i] for el in col])
+                                    new_col.info.format = col.info.format
 
                                     new_col_iter_str_vals = self.fill_values(
+
                                         col, new_col.info.iter_str_vals())
                                     col_str_iters.append(new_col_iter_str_vals)
                                     new_cols_escaped.append(col_escaped)
