diff --git a/django/utils/autoreload.py b/django/utils/autoreload.py
index b6af5f5491..70cf7758e8 100644
--- a/django/utils/autoreload.py
+++ b/django/utils/autoreload.py
@@ -220,11 +220,18 @@ def get_child_arguments():
     py_script = Path(sys.argv[0])
 
     args = [sys.executable] + ['-W%s' % o for o in sys.warnoptions]
-    # __spec__ is set when the server was started with the `-m` option,
-    # see https://docs.python.org/3/reference/import.html#main-spec
+    # If started with `python -m`, __main__.__spec__ is a ModuleSpec
+    # and spec.parent is the package. We want:
+    #  - for packages (spec.name endswith ".__main__") → spec.parent
+    #  - otherwise (nested modules) → the full spec.name
     # __spec__ may not exist, e.g. when running in a Conda env.
-    if getattr(__main__, '__spec__', None) is not None and __main__.__spec__.parent:
-        args += ['-m', __main__.__spec__.parent]
+    spec = getattr(__main__, '__spec__', None)
+    if spec is not None and spec.parent:
+        if spec.name.endswith(".__main__"):
+            module_name = spec.parent
+        else:
+            module_name = spec.name
+        args += ['-m', module_name]
         args += sys.argv[1:]
     elif not py_script.exists():
         # sys.argv[0] may not exist for several reasons on Windows.
