diff --git a/django/db/models/deletion.py b/django/db/models/deletion.py
index 92fe4affb9..05ebe3602c 100644
--- a/django/db/models/deletion.py
+++ b/django/db/models/deletion.py
@@ -276,7 +276,20 @@ class Collector:
             instance = list(instances)[0]
             if self.can_fast_delete(instance):
                 with transaction.mark_for_rollback_on_error():
+                    # Send pre_delete signal
+                    if not model._meta.auto_created:
+                        signals.pre_delete.send(
+                            sender=model, instance=instance, using=self.using
+                        )
+                    # Fast delete
                     count = sql.DeleteQuery(model).delete_batch([instance.pk], self.using)
+                    # Send post_delete signal
+                    if not model._meta.auto_created:
+                        signals.post_delete.send(
+                            sender=model, instance=instance, using=self.using
+                        )
+                    # Update the instance's PK to None
+                    setattr(instance, model._meta.pk.attname, None)
                 return count, {model._meta.label: count}
 
         with transaction.atomic(using=self.using, savepoint=False):
