diff --git b/sphinx/ext/autodoc/__init__.py a/sphinx/ext/autodoc/__init__.py
index 2b9af051d..1f4c98284 100644
--- b/sphinx/ext/autodoc/__init__.py
+++ a/sphinx/ext/autodoc/__init__.py
@@ -2013,15 +2013,12 @@ class DataDocumenter(GenericAliasMixin, NewTypeMixin, TypeVarMixin,
         self.analyzer = None
 
         if not more_content:
             more_content = StringList()
 
-        # Call the parent class's add_content method first to add the docstring
-        super().add_content(more_content, no_docstring=no_docstring)
-        
-        # Then call update_content to add the "alias of..." text
         self.update_content(more_content)
+        super().add_content(more_content, no_docstring=no_docstring)
 
 
 class NewTypeDataDocumenter(DataDocumenter):
     """
     Specialized Documenter subclass for NewTypes.
@@ -2596,14 +2593,12 @@ class AttributeDocumenter(GenericAliasMixin, NewTypeMixin, SlotsMixin,  # type:
         # AttributeDocumenter.add_content()
         self.analyzer = None
 
         if more_content is None:
             more_content = StringList()
-        # Call the parent class's add_content method first to add the docstring
-        super().add_content(more_content, no_docstring)
-        # Then call update_content to add the "alias of..." text
         self.update_content(more_content)
+        super().add_content(more_content, no_docstring)
 
 
 class PropertyDocumenter(DocstringStripSignatureMixin, ClassLevelDocumenter):  # type: ignore
     """
     Specialized Documenter subclass for properties.
