diff --git a/sphinx/cmd/quickstart.py b/sphinx/cmd/quickstart.py
index a40a21073..6c2fa35dd 100644
--- a/sphinx/cmd/quickstart.py
+++ b/sphinx/cmd/quickstart.py
@@ -89,6 +89,10 @@ class ValidationError(Exception):
 
 
 def is_path(x: str) -> str:
+    # Allow "" as an "exit" sentinel from ask_user's blank-to-exit prompt
+    if x == "":
+        return x
+    # Otherwise require an existing directory
     x = path.expanduser(x)
     if not path.isdir(x):
         raise ValidationError(__("Please enter a valid path name."))
