diff --git a/django/contrib/admin/utils.py b/django/contrib/admin/utils.py
index 07cdaa5..324836b 100644
--- a/django/contrib/admin/utils.py
+++ b/django/contrib/admin/utils.py
@@ -394,9 +394,14 @@ def display_for_field(value, field, empty_value_display):
         return formats.localize(value)
     elif isinstance(field, models.DecimalField):
         return formats.number_format(value, field.decimal_places)
+
     elif isinstance(field, (models.IntegerField, models.FloatField)):
         return formats.number_format(value)
+    elif isinstance(field, models.JSONField):
+        # Reuse the form field's rendering logic for consistency.
+        return field.formfield().prepare_value(value)
     elif isinstance(field, models.FileField) and value:
+
         return format_html('<a href="{}">{}</a>', value.url, value)
     else:
         return display_for_value(value, empty_value_display)
