diff --git a/sympy/core/numbers.py b/sympy/core/numbers.py
index 52f7ea45ad..f3d955d728 100644
--- a/sympy/core/numbers.py
+++ b/sympy/core/numbers.py
@@ -1625,11 +1625,14 @@ def __new__(cls, p, q=None, gcd=None):
             q = 1
             gcd = 1
 
         if not isinstance(p, SYMPY_INTS):
             p = Rational(p)
-            q *= p.q
+            if isinstance(q, str):
+                q = Rational(q) * p.q
+            else:
+                q *= p.q
             p = p.p
         else:
             p = int(p)
 
         if not isinstance(q, SYMPY_INTS):
