diff --git a/src/_pytest/capture.py b/src/_pytest/capture.py
index 25eab7fdf..d069f7583 100644
--- a/src/_pytest/capture.py
+++ b/src/_pytest/capture.py
@@ -446,6 +446,16 @@ class EncodedFile(object):
     def name(self):
         """Ensure that file.name is a string."""
         return repr(self.buffer)
+        
+    @property
+    def mode(self):
+        """Return non-binary mode even though buffer is binary.
+        
+        This property is needed for compatibility with code that
+        uses the presence of 'b' in the mode to determine whether
+        to pass bytes or strings.
+        """
+        return getattr(self.buffer, "mode", "").replace("b", "")
 
     def __getattr__(self, name):
         return getattr(object.__getattribute__(self, "buffer"), name)

