diff --git a/django/utils/autoreload.py b/django/utils/autoreload.py
index b8efb9f881..050f4c6b9c 100644
--- a/django/utils/autoreload.py
+++ b/django/utils/autoreload.py
@@ -225,6 +225,16 @@ def get_child_arguments():
         # The server was started with `python -m django runserver`.
         args += ['-m', 'django']
         args += sys.argv[1:]
+    elif py_script.name == '__main__.py' and hasattr(sys.modules['__main__'], '__spec__'):
+        # The server was started with `python -m <module> runserver`.
+        # Get the module name from __spec__
+        main_mod = sys.modules['__main__']
+        main_spec = main_mod.__spec__
+        modname = main_spec.name
+        if modname.endswith('.__main__'):
+            modname = modname[:-9]  # drop '.__main__'
+        args += ['-m', modname]
+        args += sys.argv[1:]
     elif not py_script.exists():
         # sys.argv[0] may not exist for several reasons on Windows.
         # It may exist with a .exe extension or have a -script.py suffix.
