diff --git a/django/utils/formats.py b/django/utils/formats.py
index 3f38322d84..ca4518ab9e 100644
--- a/django/utils/formats.py
+++ b/django/utils/formats.py
@@ -104,10 +104,11 @@ def get_format(format_type, lang=None, use_l10n=None):
     format_type is the name of the format, e.g. 'DATE_FORMAT'.
 
     If use_l10n is provided and is not None, it forces the value to
     be localized (or not), overriding the value of settings.USE_L10N.
     """
+    format_type = str(format_type)  # Convert format_type to a string
     if use_l10n is None:
         try:
             use_l10n = settings._USE_L10N_INTERNAL
         except AttributeError:
             use_l10n = settings.USE_L10N
