diff --git b/sphinx/util/inspect.py a/sphinx/util/inspect.py
index a2f5745d3..a415a7074 100644
--- b/sphinx/util/inspect.py
+++ a/sphinx/util/inspect.py
@@ -455,13 +455,10 @@ def object_description(object: Any) -> str:
         except TypeError:
             pass  # Cannot sort frozenset values, fall back to generic repr
         else:
             return "frozenset({%s})" % ", ".join(object_description(x)
                                                  for x in sorted_values)
-    if isenumattribute(object):
-        # Special case for Enum values
-        return str(object)
     try:
         s = repr(object)
     except Exception as exc:
         raise ValueError from exc
     # Strip non-deterministic memory addresses such as
