diff --git a/sympy/physics/quantum/tensorproduct.py b/sympy/physics/quantum/tensorproduct.py
index 153c783..b89cabe 100644
--- a/sympy/physics/quantum/tensorproduct.py
+++ b/sympy/physics/quantum/tensorproduct.py
@@ -2,7 +2,9 @@
 
 from __future__ import print_function, division
 
-from sympy import Expr, Add, Mul, Matrix, Pow, sympify
+
+from sympy import Expr, Add, Mul, Matrix, Pow, sympify, S
+
 from sympy.core.compatibility import range
 from sympy.core.trace import Tr
 from sympy.printing.pretty.stringpict import prettyForm
@@ -235,10 +237,22 @@ class TensorProduct(Expr):
                 s = s + '\\otimes '
         return s
 
+
     def doit(self, **hints):
         return TensorProduct(*[item.doit(**hints) for item in self.args])
 
+    def _eval_power(self, exp):
+        exp = sympify(exp)
+        if not exp.is_Integer:
+            return
+        if exp == 0:
+            return S.One
+        if exp == 1:
+            return self
+        return TensorProduct(*[arg**exp for arg in self.args])
+
     def _eval_expand_tensorproduct(self, **hints):
+
         """Distribute TensorProducts across addition."""
         args = self.args
         add_args = []
