diff --git a/sympy/polys/monomials.py b/sympy/polys/monomials.py
index 9019221..0f6d903 100644
--- a/sympy/polys/monomials.py
+++ b/sympy/polys/monomials.py
@@ -124,10 +124,12 @@ def itermonomials(variables, max_degrees, min_degrees=None):
                 powers = dict()
                 for variable in variables:
                     powers[variable] = 0
+
                 for variable in item:
                     if variable != 1:
                         powers[variable] += 1
-                if max(powers.values()) >= min_degree:
+                if sum(powers.values()) >= min_degree:
+
                     monomials_list_comm.append(Mul(*item))
             yield from set(monomials_list_comm)
         else:
@@ -136,10 +138,12 @@ def itermonomials(variables, max_degrees, min_degrees=None):
                 powers = dict()
                 for variable in variables:
                     powers[variable] = 0
+
                 for variable in item:
                     if variable != 1:
                         powers[variable] += 1
-                if max(powers.values()) >= min_degree:
+                if sum(powers.values()) >= min_degree:
+
                     monomials_list_non_comm.append(Mul(*item))
             yield from set(monomials_list_non_comm)
     else:
