diff --git a/django/utils/autoreload.py b/django/utils/autoreload.py
index cc65c043ab..164be9289c 100644
--- a/django/utils/autoreload.py
+++ b/django/utils/autoreload.py
@@ -218,10 +218,16 @@ def get_child_arguments():
     """
     import __main__
     py_script = Path(sys.argv[0])
 
     args = [sys.executable] + ['-W%s' % o for o in sys.warnoptions]
+    # Add -X options
+    for key, value in getattr(sys, '_xoptions', {}).items():
+        if value is True:
+            args.extend(['-X', key])
+        else:
+            args.extend(['-X', f'{key}={value}'])
     # __spec__ is set when the server was started with the `-m` option,
     # see https://docs.python.org/3/reference/import.html#main-spec
     # __spec__ may not exist, e.g. when running in a Conda env.
     if getattr(__main__, '__spec__', None) is not None:
         spec = __main__.__spec__
