diff --git a/django/forms/models.py b/django/forms/models.py
index 3670478..9665008 100644
--- a/django/forms/models.py
+++ b/django/forms/models.py
@@ -12,9 +12,11 @@ from django.forms.fields import ChoiceField, Field
 from django.forms.forms import BaseForm, DeclarativeFieldsMetaclass
 from django.forms.formsets import BaseFormSet, formset_factory
 from django.forms.utils import ErrorList
+
 from django.forms.widgets import (
-    HiddenInput, MultipleHiddenInput, SelectMultiple,
+    HiddenInput, MultipleHiddenInput, SelectMultiple, RadioSelect,
 )
+
 from django.utils.text import capfirst, get_text_list
 from django.utils.translation import gettext, gettext_lazy as _
 
@@ -1192,11 +1194,16 @@ class ModelChoiceField(ChoiceField):
 
         # Call Field instead of ChoiceField __init__() because we don't need
         # ChoiceField.__init__().
+
         Field.__init__(
             self, required=required, widget=widget, label=label,
             initial=initial, help_text=help_text, **kwargs
         )
+        # Skip the empty choice for widgets that can display an unselected state.
+        if required and isinstance(self.widget, RadioSelect):
+            self.empty_label = None
         self.queryset = queryset
+
         self.limit_choices_to = limit_choices_to   # limit the queryset later.
         self.to_field_name = to_field_name
 
