diff --git a/requests/auth.py b/requests/auth.py
index a3de123d..c6b8143a 100644
--- a/requests/auth.py
+++ b/requests/auth.py
@@ -122,13 +122,15 @@ class HTTPDigestAuth(AuthBase):
         cnonce = (hashlib.sha1(s).hexdigest()[:16])
         noncebit = "%s:%s:%s:%s:%s" % (nonce, ncvalue, cnonce, qop, HA2)
         if _algorithm == 'MD5-SESS':
             HA1 = hash_utf8('%s:%s:%s' % (HA1, nonce, cnonce))
 
+        chosen_qop = None
         if qop is None:
             respdig = KD(HA1, "%s:%s" % (nonce, HA2))
         elif qop == 'auth' or 'auth' in qop.split(','):
+            chosen_qop = 'auth'
             respdig = KD(HA1, noncebit)
         else:
             # XXX handle auth-int.
             return None
 
@@ -142,11 +144,11 @@ class HTTPDigestAuth(AuthBase):
         if algorithm:
             base += ', algorithm="%s"' % algorithm
         if entdig:
             base += ', digest="%s"' % entdig
         if qop:
-            base += ', qop=auth, nc=%s, cnonce="%s"' % (ncvalue, cnonce)
+            base += ', qop="%s", nc=%s, cnonce="%s"' % (chosen_qop, ncvalue, cnonce)
 
         return 'Digest %s' % (base)
 
     def handle_401(self, r, **kwargs):
         """Takes the given response and tries digest-auth, if needed."""
