diff --git a/pylint/checkers/similar.py b/pylint/checkers/similar.py
index c65b1c893..9bbf5218d 100644
--- a/pylint/checkers/similar.py
+++ b/pylint/checkers/similar.py
@@ -231,15 +231,21 @@ def hash_lineset(
     The second dict, associates the index of the starting line in the lineset's stripped lines to the
     couple [start, end] lines number in the corresponding file.
 
     :param lineset: lineset object (i.e the lines in a file)
     :param min_common_lines: number of successive lines that are used to compute the hash
+                            If set to 0, duplicate code check is disabled
     :return: a dict linking hashes to corresponding start index and a dict that links this
              index to the start and end lines in the file
     """
     hash2index = defaultdict(list)
     index2lines = {}
+    
+    # If min_common_lines is 0, disable duplicate code check by returning empty dicts
+    if min_common_lines == 0:
+        return hash2index, index2lines
+        
     # Comments, docstring and other specific patterns maybe excluded -> call to stripped_lines
     # to get only what is desired
     lines = tuple(x.text for x in lineset.stripped_lines)
     # Need different iterators on same lines but each one is shifted 1 from the precedent
     shifted_lines = [iter(lines[i:]) for i in range(min_common_lines)]
