diff --git a/django/contrib/postgres/forms/array.py b/django/contrib/postgres/forms/array.py
index 2e19cd574a..87cf4fb9e8 100644
--- a/django/contrib/postgres/forms/array.py
+++ b/django/contrib/postgres/forms/array.py
@@ -137,13 +137,14 @@ class SplitArrayWidget(forms.Widget):
             self.widget.is_localized = self.is_localized
         value = value or []
         context['widget']['subwidgets'] = []
-        final_attrs = self.build_attrs(attrs)
-        id_ = final_attrs.get('id')
+        base_attrs = self.build_attrs(attrs)
+        id_ = base_attrs.get('id')
         for i in range(max(len(value), self.size)):
             try:
                 widget_value = value[i]
             except IndexError:
                 widget_value = None
+            final_attrs = base_attrs.copy()  # Create a new copy for each widget
             if id_:
                 final_attrs = {**final_attrs, 'id': '%s_%s' % (id_, i)}
             context['widget']['subwidgets'].append(

