diff --git a/astropy/modeling/separable.py b/astropy/modeling/separable.py
index d6e9246..9e3abb9 100644
--- a/astropy/modeling/separable.py
+++ b/astropy/modeling/separable.py
@@ -238,11 +238,13 @@ def _cstack(left, right):
     else:
         cleft = np.zeros((noutp, left.shape[1]))
         cleft[: left.shape[0], : left.shape[1]] = left
+
     if isinstance(right, Model):
         cright = _coord_matrix(right, 'right', noutp)
     else:
         cright = np.zeros((noutp, right.shape[1]))
-        cright[-right.shape[0]:, -right.shape[1]:] = 1
+        cright[-right.shape[0]:, -right.shape[1]:] = right
+
 
     return np.hstack([cleft, cright])
 
