diff --git a/django/core/management/base.py b/django/core/management/base.py
index e6cbf8c64f..6fb50fca1d 100644
--- a/django/core/management/base.py
+++ b/django/core/management/base.py
@@ -149,6 +149,15 @@ class OutputWrapper(TextIOBase):
             msg += ending
         style_func = style_func or self.style_func
         self._out.write(style_func(msg))
+        
+    def flush(self):
+        """
+        Delegate to the underlying stream's flush() if it exists.
+        TextIOBase.flush() is a no-op, so we must override here.
+        """
+        flush_method = getattr(self._out, 'flush', None)
+        if callable(flush_method):
+            flush_method()
 
 
 class BaseCommand:
