diff --git a/lib/matplotlib/widgets.py b/lib/matplotlib/widgets.py
index a199e45d40..7b1a2f67f7 100644
--- a/lib/matplotlib/widgets.py
+++ b/lib/matplotlib/widgets.py
@@ -2366,9 +2366,11 @@ class ToolLineHandles:
             line_props = {}
         line_props.update({'visible': False, 'animated': useblit})
 
+        # Create line functions with autoscale=False to prevent the axis limits 
+        # from being expanded to include the handle positions
         line_fun = ax.axvline if self.direction == 'horizontal' else ax.axhline
+        
         self._line_props = line_props
-
         self.artists = [line_fun(p, **line_props) for p in positions]
 
     @property
