diff --git a/sphinx/domains/python.py b/sphinx/domains/python.py
index bcb7471..b14fcab 100644
--- a/sphinx/domains/python.py
+++ b/sphinx/domains/python.py
@@ -787,7 +787,13 @@ class PyMethod(PyObject):
             clsname, methname = name.rsplit('.', 1)
             if modname and self.env.config.add_module_names:
                 clsname = '.'.join([modname, clsname])
+
         except ValueError:
+            if 'property' in self.options:
+                if modname:
+                    return _('%s (in module %s)') % (name, modname)
+                else:
+                    return name
             if modname:
                 return _('%s() (in module %s)') % (name, modname)
             else:
@@ -796,13 +802,14 @@ class PyMethod(PyObject):
         if 'classmethod' in self.options:
             return _('%s() (%s class method)') % (methname, clsname)
         elif 'property' in self.options:
-            return _('%s() (%s property)') % (methname, clsname)
+            return _('%s (%s property)') % (methname, clsname)
         elif 'staticmethod' in self.options:
             return _('%s() (%s static method)') % (methname, clsname)
         else:
             return _('%s() (%s method)') % (methname, clsname)
 
 
+
 class PyClassMethod(PyMethod):
     """Description of a classmethod."""
 
