diff --git a/src/_pytest/logging.py b/src/_pytest/logging.py
index 16efd55..2eea0c5 100644
--- a/src/_pytest/logging.py
+++ b/src/_pytest/logging.py
@@ -344,8 +344,12 @@ class LogCaptureFixture:
     def __init__(self, item: nodes.Node) -> None:
         """Creates a new funcarg."""
         self._item = item
+
         # dict of log name -> log level
         self._initial_logger_levels = {}  # type: Dict[Optional[str], int]
+        # Remember the original handler level to restore it during fixture finalization.
+        self._initial_handler_level = self.handler.level
+
 
     def _finalize(self) -> None:
         """Finalizes the fixture.
@@ -353,9 +357,13 @@ class LogCaptureFixture:
         This restores the log levels changed by :meth:`set_level`.
         """
         # restore log levels
+
         for logger_name, level in self._initial_logger_levels.items():
             logger = logging.getLogger(logger_name)
             logger.setLevel(level)
+        # restore the handler's original level
+        self.handler.setLevel(self._initial_handler_level)
+
 
     @property
     def handler(self) -> LogCaptureHandler:
