diff --git a/requests/adapters.py b/requests/adapters.py
index fe22ff45..29c241fa 100644
--- a/requests/adapters.py
+++ b/requests/adapters.py
@@ -27,7 +27,7 @@ from urllib3.exceptions import ProtocolError
 from urllib3.exceptions import ReadTimeoutError
 from urllib3.exceptions import SSLError as _SSLError
 from urllib3.exceptions import ResponseError
-from urllib3.exceptions import LocationValueError
+from urllib3.exceptions import LocationValueError, LocationParseError
 
 from .models import Response
 from .compat import urlparse, basestring
@@ -497,6 +497,11 @@ class HTTPAdapter(BaseAdapter):
                     low_conn.close()
                     raise
 
+        except LocationParseError as e:
+            # urllib3 can notice an invalid host name during connection
+            # set-up and raise LocationParseError; translate this to InvalidURL
+            # for users, like we already do for LocationValueError earlier.
+            raise InvalidURL(*e.args, request=request)
         except (ProtocolError, socket.error) as err:
             raise ConnectionError(err, request=request)
 
