diff --git a/sklearn/utils/_set_output.py b/sklearn/utils/_set_output.py
index 335773c6a..fdbd2a977 100644
--- a/sklearn/utils/_set_output.py
+++ b/sklearn/utils/_set_output.py
@@ -54,11 +54,13 @@ def _wrap_in_pandas_container(
 
     if isinstance(data_to_wrap, pd.DataFrame):
         if columns is not None:
             data_to_wrap.columns = columns
         if index is not None:
-            data_to_wrap.index = index
+            # Only set the index if the lengths match
+            if len(data_to_wrap) == len(index):
+                data_to_wrap.index = index
         return data_to_wrap
 
     return pd.DataFrame(data_to_wrap, index=index, columns=columns)
 
 
