--- a/django/forms/models.py
+++ b/django/forms/models.py
@@ -1177,7 +1177,10 @@
                 to_field = self.instance._meta.get_field(kwargs["to_field"])
             else:
                 to_field = self.instance._meta.pk
-            if to_field.has_default():
+            if to_field.has_default() and (
+                to_field.get_internal_type() in ('AutoField', 'BigAutoField', 'SmallAutoField') or
+                (to_field.primary_key and to_field.get_internal_type() == 'UUIDField')
+            ):
                 setattr(self.instance, to_field.attname, None)
 
         form.fields[name] = InlineForeignKeyField(self.instance, **kwargs)
