diff --git a/astropy/table/table.py b/astropy/table/table.py
index 5afe2127a3..7ab281d5c9 100644
--- a/astropy/table/table.py
+++ b/astropy/table/table.py
@@ -1241,10 +1241,15 @@ class Table:
 
         # Structured ndarray gets viewed as a mixin unless already a valid
         # mixin class
         if (not isinstance(data, Column) and not data_is_mixin
                 and isinstance(data, np.ndarray) and len(data.dtype) > 1):
+            warnings.warn("Automatic conversion of structured numpy arrays to "
+                         "NdarrayMixin is deprecated and will be removed in a "
+                         "future version. To avoid this warning, explicitly "
+                         "wrap the array in a Column: Column(array).",
+                         FutureWarning, stacklevel=4)
             data = data.view(NdarrayMixin)
             data_is_mixin = True
 
         # Get the final column name using precedence.  Some objects may not
         # have an info attribute. Also avoid creating info as a side effect.
diff --git a/setup.cfg b/setup.cfg
index 3503fd9027..d7d193c996 100644
--- a/setup.cfg
+++ b/setup.cfg
@@ -136,10 +136,11 @@ filterwarnings =
     ignore:Importing from numpy:DeprecationWarning:scipy
     ignore:Conversion of the second argument:FutureWarning:scipy
     ignore:Using a non-tuple sequence:FutureWarning:scipy
     ignore:Using or importing the ABCs from 'collections':DeprecationWarning
     ignore:Unknown pytest\.mark\.mpl_image_compare:pytest.PytestUnknownMarkWarning
+    ignore:Automatic conversion of structured numpy arrays to NdarrayMixin is deprecated:FutureWarning
     ignore:Unknown config option:pytest.PytestConfigWarning
     ignore:matplotlibrc text\.usetex:UserWarning:matplotlib
     ignore:The 'text' argument to find\(\)-type methods is deprecated:DeprecationWarning
     # Triggered by ProgressBar > ipykernel.iostream
     ignore:the imp module is deprecated:DeprecationWarning
