diff --git a/requests/adapters.py b/requests/adapters.py
index fe22ff45..0a9efd17 100644
--- a/requests/adapters.py
+++ b/requests/adapters.py
@@ -26,10 +26,11 @@ from urllib3.exceptions import ProxyError as _ProxyError
 from urllib3.exceptions import ProtocolError
 from urllib3.exceptions import ReadTimeoutError
 from urllib3.exceptions import SSLError as _SSLError
 from urllib3.exceptions import ResponseError
 from urllib3.exceptions import LocationValueError
+from urllib3.exceptions import LocationParseError
 
 from .models import Response
 from .compat import urlparse, basestring
 from .utils import (DEFAULT_CA_BUNDLE_PATH, extract_zipped_paths,
                     get_encoding_from_headers, prepend_scheme_if_needed,
@@ -495,10 +496,13 @@ class HTTPAdapter(BaseAdapter):
                     # If we hit any problems here, clean up the connection.
                     # Then, reraise so that we can handle the actual exception.
                     low_conn.close()
                     raise
 
+        except LocationParseError as e:
+            raise InvalidURL(e, request=request)
+
         except (ProtocolError, socket.error) as err:
             raise ConnectionError(err, request=request)
 
         except MaxRetryError as e:
             if isinstance(e.reason, ConnectTimeoutError):
@@ -533,6 +537,8 @@ class HTTPAdapter(BaseAdapter):
             elif isinstance(e, _InvalidHeader):
                 raise InvalidHeader(e, request=request)
             else:
                 raise
 
+
+
         return self.build_response(request, resp)
