diff --git a/sklearn/linear_model/logistic.py b/sklearn/linear_model/logistic.py
index e4ea696ce..c49e732f9 100644
--- a/sklearn/linear_model/logistic.py
+++ b/sklearn/linear_model/logistic.py
@@ -922,7 +922,20 @@ def _log_reg_scoring_path(X, y, train, test, pos_class=None, Cs=10,
         check_input=False, max_squared_sum=max_squared_sum,
         sample_weight=sample_weight)
 
-    log_reg = LogisticRegression(fit_intercept=fit_intercept)
+    log_reg = LogisticRegression(
+        fit_intercept=fit_intercept,
+        multi_class=multi_class,
+        solver=solver,
+        penalty=penalty,
+        dual=dual,
+        tol=tol,
+        C=1.0,  # C value doesn't matter as we set coef_ directly
+        intercept_scaling=float(intercept_scaling),
+        class_weight=class_weight,
+        max_iter=max_iter,
+        random_state=random_state,
+        verbose=verbose
+    )
 
     # The score method of Logistic Regression has a classes_ attribute.
     if multi_class == 'ovr':

