diff --git a/django/db/models/enums.py b/django/db/models/enums.py
index f48143ddaf..c2f1dde80e 100644
--- a/django/db/models/enums.py
+++ b/django/db/models/enums.py
@@ -29,10 +29,13 @@ class ChoicesMeta(enum.EnumMeta):
         cls._value2label_map_ = dict(zip(cls._value2member_map_, labels))
         # Add a label property to instances of enum which uses the enum member
         # that is passed in as "self" as the value to use when looking up the
         # label in the choices.
         cls.label = property(lambda self: cls._value2label_map_.get(self.value))
+        # Prevent Django templates from calling the class (which would fail
+        # because it requires a value argument).
+        cls.do_not_call_in_templates = True
         return enum.unique(cls)
 
     def __contains__(cls, member):
         if not isinstance(member, enum.Enum):
             # Allow non-enums to match against member values.
