diff --git a/lib/matplotlib/__init__.py b/lib/matplotlib/__init__.py
index c268a56724..2cbdf74b2f 100644
--- a/lib/matplotlib/__init__.py
+++ b/lib/matplotlib/__init__.py
@@ -1173,11 +1173,11 @@ def get_backend():
 
     See Also
     --------
     matplotlib.use
     """
-    return rcParams['backend']
+    return dict.__getitem__(rcParams, 'backend')
 
 
 def interactive(b):
     """
     Set whether to redraw after every plotting command (e.g. `.pyplot.xlabel`).
