--- a/django/contrib/admin/options.py
+++ b/django/contrib/admin/options.py
@@ -1912,6 +1912,10 @@
 
     @csrf_protect_m
     def changelist_view(self, request, extra_context=None):
+        with transaction.atomic(using=router.db_for_write(self.model)):
+            return self._changelist_view(request, extra_context)
+
+    def _changelist_view(self, request, extra_context=None):
         """
         The 'change list' admin view for this model.
         """
