diff --git a/django/db/models/base.py b/django/db/models/base.py
index 4c7a87c..e46ebb5 100644
--- a/django/db/models/base.py
+++ b/django/db/models/base.py
@@ -1711,8 +1711,12 @@ class Model(metaclass=ModelBase):
                     fld = _cls._meta.get_field(part)
                     if fld.is_relation:
                         _cls = fld.get_path_info()[-1].to_opts.model
+
                 except (FieldDoesNotExist, AttributeError):
+                    if part == 'pk':
+                        continue
                     if fld is None or fld.get_transform(part) is None:
+
                         errors.append(
                             checks.Error(
                                 "'ordering' refers to the nonexistent field, "
