diff --git a/django/db/models/aggregates.py b/django/db/models/aggregates.py
index 8c4eae7906..acbd2e4879 100644
--- a/django/db/models/aggregates.py
+++ b/django/db/models/aggregates.py
@@ -63,11 +63,13 @@ class Aggregate(Func):
         if (default := c.default) is None:
             return c
         if hasattr(default, 'resolve_expression'):
             default = default.resolve_expression(query, allow_joins, reuse, summarize)
         c.default = None  # Reset the default argument before wrapping.
-        return Coalesce(c, default, output_field=c._output_field_or_none)
+        coalesce = Coalesce(c, default, output_field=c._output_field_or_none)
+        coalesce.is_summary = summarize
+        return coalesce
 
     @property
     def default_alias(self):
         expressions = self.get_source_expressions()
         if len(expressions) == 1 and hasattr(expressions[0], 'name'):
