diff --git a/sympy/core/basic.py b/sympy/core/basic.py
index a947e2a..17b9a65 100644
--- a/sympy/core/basic.py
+++ b/sympy/core/basic.py
@@ -312,8 +312,10 @@ class Basic(with_metaclass(ManagedProperties)):
                 return self == other.base
             try:
                 other = _sympify(other)
+
             except SympifyError:
-                return False    # sympy != other
+                return NotImplemented    # sympy != other
+
 
             if type(self) != type(other):
                 return False
@@ -328,8 +330,13 @@ class Basic(with_metaclass(ManagedProperties)):
              a.compare(b) != 0
 
            but faster
+
         """
-        return not self.__eq__(other)
+        result = self.__eq__(other)
+        if result is NotImplemented:
+            return NotImplemented
+        return not result
+
 
     def dummy_eq(self, other, symbol=None):
         """
