diff --git a/django/forms/fields.py b/django/forms/fields.py
index 65d6a9ec82..cd91b97ff6 100644
--- a/django/forms/fields.py
+++ b/django/forms/fields.py
@@ -1024,10 +1024,18 @@ class MultiValueField(Field):
         if not value or isinstance(value, (list, tuple)):
             if not value or not [v for v in value if v not in self.empty_values]:
                 if self.required:
                     raise ValidationError(self.error_messages['required'], code='required')
                 else:
+                    # Check if any individual fields are required
+                    errors = []
+                    for i, field in enumerate(self.fields):
+                        if field.required:
+                            if field.error_messages['incomplete'] not in errors:
+                                errors.append(field.error_messages['incomplete'])
+                    if errors:
+                        raise ValidationError(errors)
                     return self.compress([])
         else:
             raise ValidationError(self.error_messages['invalid'], code='invalid')
         for i, field in enumerate(self.fields):
             try:
