diff --git a/django/db/models/expressions.py b/django/db/models/expressions.py
index c1a76584f0..aface75cff 100644
--- a/django/db/models/expressions.py
+++ b/django/db/models/expressions.py
@@ -861,10 +861,13 @@ class ExpressionWrapper(Expression):
         self.expression = exprs[0]
 
     def get_source_expressions(self):
         return [self.expression]
 
+    def get_group_by_cols(self, alias=None):
+        return self.expression.get_group_by_cols(alias)
+
     def as_sql(self, compiler, connection):
         return self.expression.as_sql(compiler, connection)
 
     def __repr__(self):
         return "{}({})".format(self.__class__.__name__, self.expression)
