diff --git a/xarray/core/variable.py b/xarray/core/variable.py
index 5827b90a..1941e1b8 100644
--- a/xarray/core/variable.py
+++ b/xarray/core/variable.py
@@ -2879,11 +2879,13 @@ class IndexVariable(Variable):
     def _data_equals(self, other):
         return self.to_index().equals(other.to_index())
 
     def to_index_variable(self):
         """Return this variable as an xarray.IndexVariable"""
-        return self
+        return IndexVariable(
+            self.dims, self._data, self._attrs, encoding=self._encoding, fastpath=True
+        )
 
     to_coord = utils.alias(to_index_variable, "to_coord")
 
     def to_index(self):
         """Convert this variable to a pandas.Index"""
