diff --git b/sphinx/application.py a/sphinx/application.py
index 776db7b51..385b74d8a 100644
--- b/sphinx/application.py
+++ a/sphinx/application.py
@@ -291,11 +291,11 @@ class Sphinx:
                                      self.config.language, self.config.source_encoding)
             for catalog in repo.catalogs:
                 if catalog.domain == 'sphinx' and catalog.is_outdated():
                     catalog.write_mo(self.config.language)
 
-            locale_dirs = list(repo.locale_dirs) + [path.join(package_dir, 'locale'), None]
+            locale_dirs = [None, path.join(package_dir, 'locale')] + list(repo.locale_dirs)
             self.translator, has_translation = locale.init(locale_dirs, self.config.language)
             if has_translation or self.config.language == 'en':
                 # "en" never needs to be translated
                 logger.info(__('done'))
             else:
