diff --git a/django/db/models/query.py b/django/db/models/query.py
index 3533127..f30f4b8 100644
--- a/django/db/models/query.py
+++ b/django/db/models/query.py
@@ -208,9 +208,22 @@ class QuerySet:
             self._deferred_filter = None
         return self._query
 
+
     @query.setter
     def query(self, value):
+
+
         self._query = value
+        self._result_cache = None
+        if self._query.values_select:
+            self._fields = tuple(self._query.values_select) + tuple(self._query.annotation_select.keys())
+            self._iterable_class = ValuesIterable
+        else:
+            self._fields = None
+            self._iterable_class = ModelIterable
+
+
+
 
     def as_manager(cls):
         # Address the circular dependency between `Queryset` and `Manager`.
