diff --git a/lib/matplotlib/patches.py b/lib/matplotlib/patches.py
index 92dc55940b..8587ce3202 100644
--- a/lib/matplotlib/patches.py
+++ b/lib/matplotlib/patches.py
@@ -913,11 +913,11 @@ class RegularPolygon(Patch):
         **kwargs
             `Patch` properties:
 
             %(Patch:kwdoc)s
         """
-        self.xy = xy
+        self.xy = np.array(xy)
         self.numvertices = numVertices
         self.orientation = orientation
         self.radius = radius
         self._path = Path.unit_regular_polygon(numVertices)
         self._patch_transform = transforms.Affine2D()
diff --git a/lib/matplotlib/text.py b/lib/matplotlib/text.py
index d4922fc517..85134e9cc4 100644
--- a/lib/matplotlib/text.py
+++ b/lib/matplotlib/text.py
@@ -1454,11 +1454,11 @@ class _AnnotationBase:
     def __init__(self,
                  xy,
                  xycoords='data',
                  annotation_clip=None):
 
-        self.xy = xy
+        self.xy = np.array(xy)
         self.xycoords = xycoords
         self.set_annotation_clip(annotation_clip)
 
         self._draggable = None
 
