diff --git b/sphinx/domains/std.py a/sphinx/domains/std.py
index de72d568c..d820cfe5c 100644
--- b/sphinx/domains/std.py
+++ a/sphinx/domains/std.py
@@ -652,17 +652,12 @@ class StandardDomain(Domain):
 
         .. versionadded:: 3.0
         """
         if (objtype, name) in self.objects:
             docname = self.objects[objtype, name][0]
-            # For glossary terms, clarify that matching is case-insensitive
-            if objtype == 'term':
-                logger.warning(__('duplicate %s description of %s (case insensitive), other instance in %s'),
-                               objtype, name, docname, location=location)
-            else:
-                logger.warning(__('duplicate %s description of %s, other instance in %s'),
-                               objtype, name, docname, location=location)
+            logger.warning(__('duplicate %s description of %s, other instance in %s'),
+                           objtype, name, docname, location=location)
         self.objects[objtype, name] = (self.env.docname, labelid)
 
     def add_object(self, objtype: str, name: str, docname: str, labelid: str) -> None:
         warnings.warn('StandardDomain.add_object() is deprecated.',
                       RemovedInSphinx50Warning)
diff --git b/tests/test_domain_std.py a/tests/test_domain_std.py
index a4b2c0735..aa1d29eb1 100644
--- b/tests/test_domain_std.py
+++ a/tests/test_domain_std.py
@@ -182,16 +182,14 @@ def test_glossary_warning(app, status, warning):
     text = (".. glossary::\n"
             "\n"
             "   term-case4\n"
             "   term-case4\n")
     restructuredtext.parse(app, text, "case4")
-    assert ("case4.rst:3: WARNING: duplicate term description of term-case4 (case insensitive), "
+    assert ("case4.rst:3: WARNING: duplicate term description of term-case4, "
             "other instance in case4" in warning.getvalue())
 
 
-
-
 def test_glossary_comment(app):
     text = (".. glossary::\n"
             "\n"
             "   term1\n"
             "       description\n"
