diff --git a/sympy/utilities/lambdify.py b/sympy/utilities/lambdify.py
index 9dd6fe4..236963e 100644
--- a/sympy/utilities/lambdify.py
+++ b/sympy/utilities/lambdify.py
@@ -959,9 +959,14 @@ def _recursive_to_string(doprint, arg):
             left, right = "[]"
         elif isinstance(arg, tuple):
             left, right = "()"
+
         else:
             raise NotImplementedError("unhandled type: %s, %s" % (type(arg), arg))
-        return left +', '.join(_recursive_to_string(doprint, e) for e in arg) + right
+        inner = ', '.join(_recursive_to_string(doprint, e) for e in arg)
+        if isinstance(arg, tuple) and len(arg) == 1:
+            inner += ','
+        return left + inner + right
+
     elif isinstance(arg, str):
         return arg
     else:
