diff --git a/xarray/core/dataarray.py b/xarray/core/dataarray.py
index b0df8749..2aab5b54 100644
--- a/xarray/core/dataarray.py
+++ b/xarray/core/dataarray.py
@@ -1959,11 +1959,11 @@ class DataArray(AbstractArray, DataWithCoords):
         variable_dim = idx.names[level_number]
 
         # pull variables out of datarray
         data_dict = {}
         for k in variables:
-            data_dict[k] = self.sel({variable_dim: k}).squeeze(drop=True)
+            data_dict[k] = self.sel({variable_dim: k}).squeeze(drop=True).drop_vars(dim, errors="ignore")
 
         # unstacked dataset
         return Dataset(data_dict)
 
     def transpose(self, *dims: Hashable, transpose_coords: bool = True) -> "DataArray":
