diff --git a/django/db/models/expressions.py b/django/db/models/expressions.py
index 81f8f79c71..5ce19dceed 100644
--- a/django/db/models/expressions.py
+++ b/django/db/models/expressions.py
@@ -1211,16 +1211,24 @@ class Exists(Subquery):
 
     def as_sql(self, compiler, connection, template=None, **extra_context):
         query = self.query.exists(using=connection.alias)
-        sql, params = super().as_sql(
-            compiler,
-            connection,
-            template=template,
-            query=query,
-            **extra_context,
-        )
-        if self.negated:
-            sql = 'NOT {}'.format(sql)
-        return sql, params
+        try:
+            sql, params = super().as_sql(
+                compiler,
+                connection,
+                template=template,
+                query=query,
+                **extra_context,
+            )
+            if self.negated:
+                sql = 'NOT {}'.format(sql)
+            return sql, params
+        except EmptyResultSet:
+            if self.negated:
+                # An empty EXISTS subquery with NOT in front of it is TRUE
+                return '1=1', []
+            else:
+                # An empty EXISTS subquery is FALSE
+                return '0=1', []
 
     def select_format(self, compiler, sql, params):
         # Wrap EXISTS() with a CASE WHEN expression if a database backend

