diff --git a/pylint/lint/pylinter.py b/pylint/lint/pylinter.py
index 440e6fe..644c306 100644
--- a/pylint/lint/pylinter.py
+++ b/pylint/lint/pylinter.py
@@ -597,12 +597,14 @@ class PyLinter(
                         # Skip subtree of already discovered package.
                         continue
 
+
                     if _is_ignored_file(
                         root,
                         self.config.ignore,
                         self.config.ignore_patterns,
-                        self.config.ignore_paths,
+                        self._ignore_paths,
                     ):
+
                         skip_subtrees.append(root)
                         continue
 
