diff --git a/lib/matplotlib/__init__.py b/lib/matplotlib/__init__.py
index afdb2d9..441eedc 100644
--- a/lib/matplotlib/__init__.py
+++ b/lib/matplotlib/__init__.py
@@ -1138,9 +1138,15 @@ def use(backend, *, force=True):
     if rcParams._get_backend_or_none() == name:
         # Nothing to do if the requested backend is already set
         pass
+
     else:
+        current = rcsetup.validate_backend(rcParams['backend'])
+        if not force and current == name:
+            return
         # if pyplot is not already imported, do not import it.  Doing
         # so may trigger a `plt.switch_backend` to the _default_ backend
+
+
         # before we get a chance to change to the one the user just requested
         plt = sys.modules.get('matplotlib.pyplot')
         # if pyplot is imported, then try to change backends
