diff --git b/sphinx/domains/python.py a/sphinx/domains/python.py
index bf7d18cce..79d7e4f46 100644
--- b/sphinx/domains/python.py
+++ a/sphinx/domains/python.py
@@ -270,16 +270,10 @@ class PyXrefMixin:
                   innernode: "Type[TextlikeNode]" = nodes.emphasis,
                   contnode: Node = None, env: BuildEnvironment = None) -> Node:
         result = super().make_xref(rolename, domain, target,  # type: ignore
                                    innernode, contnode, env)
         result['refspecific'] = True
-        
-        # Set module and class context from the environment
-        if env:
-            result['py:module'] = env.ref_context.get('py:module')
-            result['py:class'] = env.ref_context.get('py:class')
-            
         if target.startswith(('.', '~')):
             prefix, result['reftarget'] = target[0], target[1:]
             if prefix == '.':
                 text = target[1:]
             elif prefix == '~':
