diff --git a/django/middleware/security.py b/django/middleware/security.py
index 035c329efb..8bfd615dae 100644
--- a/django/middleware/security.py
+++ b/django/middleware/security.py
@@ -19,7 +19,7 @@ class SecurityMiddleware(MiddlewareMixin):
         self.redirect_host = settings.SECURE_SSL_HOST
         self.redirect_exempt = [re.compile(r) for r in settings.SECURE_REDIRECT_EXEMPT]
         self.referrer_policy = settings.SECURE_REFERRER_POLICY
-        self.get_response = get_response
+        super().__init__(get_response)
 
     def process_request(self, request):
         path = request.path.lstrip("/")
diff --git a/django/utils/deprecation.py b/django/utils/deprecation.py
index b2c681b33c..f85fd1e9c1 100644
--- a/django/utils/deprecation.py
+++ b/django/utils/deprecation.py
@@ -106,6 +106,8 @@ class MiddlewareMixin:
             # Mark the class as async-capable, but do the actual switch
             # inside __call__ to avoid swapping out dunder methods
             self._is_coroutine = asyncio.coroutines._is_coroutine
+            # Replace __call__ with __acall__ to make this instance truly async
+            self.__call__ = self.__acall__
 
     def __call__(self, request):
         # Exit out to async mode, if needed
