diff --git b/sphinx/domains/std.py a/sphinx/domains/std.py
index edc110d2a..d820cfe5c 100644
--- b/sphinx/domains/std.py
+++ a/sphinx/domains/std.py
@@ -303,11 +303,11 @@ def make_glossary_term(env: "BuildEnvironment", textnodes: Iterable[Node], index
             node_id = 'term-%d' % env.new_serialno('glossary')
         gloss_entries.add(node_id)
         term['ids'].append(node_id)
 
     std = cast(StandardDomain, env.get_domain('std'))
-    std.note_object('term', termtext, node_id, location=term)
+    std.note_object('term', termtext.lower(), node_id, location=term)
 
     # add an index entry too
     indexnode = addnodes.index()
     indexnode['entries'] = [('single', termtext, node_id, 'main', index_key)]
     indexnode.source, indexnode.line = term.source, term.line
