diff --git a/django/template/defaultfilters.py b/django/template/defaultfilters.py
index 5ccef38048..9485539292 100644
--- a/django/template/defaultfilters.py
+++ b/django/template/defaultfilters.py
@@ -83,10 +83,12 @@ def escapejs_filter(value):
 
 
 @register.filter(is_safe=True)
-def json_script(value, element_id):
+def json_script(value, element_id=None):
     """
     Output value JSON-encoded, wrapped in a <script type="application/json">
     tag.
+    
+    If element_id is provided, it will be used as the id attribute of the script tag.
     """
     return _json_script(value, element_id)
 
diff --git a/django/utils/html.py b/django/utils/html.py
index da1b5675ec..c82120302f 100644
--- a/django/utils/html.py
+++ b/django/utils/html.py
@@ -61,7 +61,7 @@ _json_script_escapes = {
 }
 
 
-def json_script(value, element_id):
+def json_script(value, element_id=None):
     """
     Escape all the HTML/XML special characters with their unicode escapes, so
     value is safe to be output anywhere except for inside a tag attribute. Wrap
@@ -69,9 +69,14 @@ def json_script(value, element_id):
     """
     from django.core.serializers.json import DjangoJSONEncoder
     json_str = json.dumps(value, cls=DjangoJSONEncoder).translate(_json_script_escapes)
+    if element_id:
+        return format_html(
+            '<script id="{}" type="application/json">{}</script>',
+            element_id, mark_safe(json_str)
+        )
     return format_html(
-        '<script id="{}" type="application/json">{}</script>',
-        element_id, mark_safe(json_str)
+        '<script type="application/json">{}</script>',
+        mark_safe(json_str)
     )
 
 
diff --git a/docs/ref/templates/builtins.txt b/docs/ref/templates/builtins.txt
index 2edfd65c7b..0b9ae02e6c 100644
--- a/docs/ref/templates/builtins.txt
+++ b/docs/ref/templates/builtins.txt
@@ -1832,7 +1832,7 @@ If ``value`` is the list ``['a', 'b', 'c']``, the output will be the string
 Safely outputs a Python object as JSON, wrapped in a ``<script>`` tag, ready
 for use with JavaScript.
 
-**Argument:** HTML "id" of the ``<script>`` tag.
+**Argument:** Optional HTML "id" of the ``<script>`` tag.
 
 For example::
 
@@ -1844,6 +1844,18 @@ If ``value`` is the dictionary ``{'hello': 'world'}``, the output will be:
 
     <script id="hello-data" type="application/json">{"hello": "world"}</script>
 
+If no id is provided, for example:
+
+.. code-block:: html+django
+
+    {{ value|json_script }}
+
+The output will be:
+
+.. code-block:: html
+
+    <script type="application/json">{"hello": "world"}</script>
+
 The resulting data can be accessed in JavaScript like this:
 
 .. code-block:: javascript
diff --git a/tests/template_tests/filter_tests/test_json_script.py b/tests/template_tests/filter_tests/test_json_script.py
index 061fe32c12..19e4f6b9c0 100644
--- a/tests/template_tests/filter_tests/test_json_script.py
+++ b/tests/template_tests/filter_tests/test_json_script.py
@@ -17,3 +17,16 @@ class JsonScriptTests(SimpleTestCase):
             '{"a": "testing\\r\\njson \'string\\" \\u003Cb\\u003Eescaping\\u003C/b\\u003E"}'
             '</script>'
         )
+        
+    @setup({'json-tag02': '{{ value|json_script }}'})
+    def test_without_id(self):
+        output = self.engine.render_to_string(
+            'json-tag02',
+            {'value': {'a': 'testing\r\njson \'string" <b>escaping</b>'}}
+        )
+        self.assertEqual(
+            output,
+            '<script type="application/json">'
+            '{"a": "testing\\r\\njson \'string\\" \\u003Cb\\u003Eescaping\\u003C/b\\u003E"}'
+            '</script>'
+        )
diff --git a/tests/utils_tests/test_html.py b/tests/utils_tests/test_html.py
index 909620ea3f..7444839006 100644
--- a/tests/utils_tests/test_html.py
+++ b/tests/utils_tests/test_html.py
@@ -172,6 +172,28 @@ class TestUtilsHtml(SimpleTestCase):
         for arg, expected in tests:
             with self.subTest(arg=arg):
                 self.assertEqual(json_script(arg, 'test_id'), expected)
+                
+    def test_json_script_without_id(self):
+        tests = (
+            # "<", ">" and "&" are quoted inside JSON strings
+            (('&<>', '<script type="application/json">"\\u0026\\u003C\\u003E"</script>')),
+            # "<", ">" and "&" are quoted inside JSON objects
+            (
+                {'a': '<script>test&ing</script>'},
+                '<script type="application/json">'
+                '{"a": "\\u003Cscript\\u003Etest\\u0026ing\\u003C/script\\u003E"}</script>'
+            ),
+            # Lazy strings are quoted
+            (lazystr('&<>'), '<script type="application/json">"\\u0026\\u003C\\u003E"</script>'),
+            (
+                {'a': lazystr('<script>test&ing</script>')},
+                '<script type="application/json">'
+                '{"a": "\\u003Cscript\\u003Etest\\u0026ing\\u003C/script\\u003E"}</script>'
+            ),
+        )
+        for arg, expected in tests:
+            with self.subTest(arg=arg):
+                self.assertEqual(json_script(arg), expected)
 
     def test_smart_urlquote(self):
         items = (
