diff --git a/lib/matplotlib/offsetbox.py b/lib/matplotlib/offsetbox.py
index efa34dd078..ec619e416e 100644
--- a/lib/matplotlib/offsetbox.py
+++ b/lib/matplotlib/offsetbox.py
@@ -1448,6 +1448,22 @@ or callable, default: value of *xycoords*
 
     def draw(self, renderer):
         # docstring inherited
+        # Propagate *gid* and *url* to the children for the draw call
+        parent_gid = self.get_gid()
+        parent_url = self.get_url()
+        def _tmp_set(artist):
+            if artist is None:
+                return None
+            old = artist.get_gid(), artist.get_url()
+            if parent_gid is not None:
+                artist.set_gid(parent_gid)
+            if parent_url is not None:
+                artist.set_url(parent_url)
+            return old
+
+        olds = [_tmp_set(a) for a in (self.offsetbox,
+                                      self.patch,
+                                      self.arrow_patch)]
         if renderer is not None:
             self._renderer = renderer
         if not self.get_visible() or not self._check_xy(renderer):
@@ -1461,6 +1477,16 @@ or callable, default: value of *xycoords*
         self.offsetbox.draw(renderer)
         self.stale = False
 
+        # Restore previous gid/url
+        for art, old in zip((self.offsetbox,
+                             self.patch,
+                             self.arrow_patch), olds):
+            if old is None:      # artist was None
+                continue
+            gid_old, url_old = old
+            art.set_gid(gid_old)
+            art.set_url(url_old)
+
 
 class DraggableBase:
     """
