diff --git a/sympy/simplify/fu.py b/sympy/simplify/fu.py
index 0cdedc1..d16ed8f 100644
--- a/sympy/simplify/fu.py
+++ b/sympy/simplify/fu.py
@@ -498,9 +498,15 @@ def _TR56(rv, f, g, h, max, pow):
         # or only those expressible as powers of 2. Also, should it only
         # make the changes in powers that appear in sums -- making an isolated
         # change is not going to allow a simplification as far as I can tell.
+
         if not (rv.is_Pow and rv.base.func == f):
             return rv
 
+        exp = rv.exp
+        if not (exp.is_number and exp.is_Integer):
+            return rv
+
+
         if (rv.exp < 0) == True:
             return rv
         if (rv.exp > max) == True:
