diff --git a/sympy/matrices/sparse.py b/sympy/matrices/sparse.py
index e6f36cdf35..0c5d37489e 100644
--- a/sympy/matrices/sparse.py
+++ b/sympy/matrices/sparse.py
@@ -983,11 +983,11 @@ def col_join(self, other):
         of the matrix:
 
         >>> C == A.row_insert(A.rows, Matrix(B))
         True
         """
-        if not self:
+        if self.rows == 0:
             return type(self)(other)
         A, B = self, other
         if not A.cols == B.cols:
             raise ShapeError()
         A = A.copy()
@@ -1189,11 +1189,11 @@ def row_join(self, other):
         of the matrix:
 
         >>> C == A.col_insert(A.cols, B)
         True
         """
-        if not self:
+        if self.cols == 0:
             return type(self)(other)
         A, B = self, other
         if not A.rows == B.rows:
             raise ShapeError()
         A = A.copy()
