--- a/lib/matplotlib/offsetbox.py
+++ b/lib/matplotlib/offsetbox.py
@@ -166,12 +166,18 @@
         descent = max(d for h, d in hd_list)
         height = height_descent + descent
         offsets = [0. for h, d in hd_list]
-    elif align in ["left", "top"]:
+    elif align == "left":
         descent = 0.
         offsets = [d for h, d in hd_list]
-    elif align in ["right", "bottom"]:
+    elif align == "right":
         descent = 0.
         offsets = [height - h + d for h, d in hd_list]
+    elif align == "top":
+        descent = 0.
+        offsets = [height - h + d for h, d in hd_list]
+    elif align == "bottom":
+        descent = 0.
+        offsets = [d for h, d in hd_list]
     elif align == "center":
         descent = 0.
         offsets = [(height - h) * .5 + d for h, d in hd_list]
