diff --git a/src/_pytest/logging.py b/src/_pytest/logging.py
index a4f4214b1..5d5844e8c 100644
--- a/src/_pytest/logging.py
+++ b/src/_pytest/logging.py
@@ -439,10 +439,13 @@ class LogCaptureFixture:
         return [r.getMessage() for r in self.records]
 
     def clear(self) -> None:
         """Reset the list of log records and the captured log text."""
         self.handler.reset()
+        # Update the stash to point to the new empty list
+        for when in self._item.stash.get(caplog_records_key, {}):
+            self._item.stash[caplog_records_key][when] = self.handler.records
 
     def set_level(self, level: Union[int, str], logger: Optional[str] = None) -> None:
         """Set the level of a logger for the duration of a test.
 
         .. versionchanged:: 3.4
