diff --git a/django/forms/models.py b/django/forms/models.py
index 5edbbd376f..6bd72ac592 100644
--- a/django/forms/models.py
+++ b/django/forms/models.py
@@ -42,15 +42,22 @@ def construct_instance(form, instance, fields=None, exclude=None):
         if not f.editable or isinstance(f, models.AutoField) \
                 or f.name not in cleaned_data:
             continue
-        if fields is not None and f.name not in fields:
+
+        if fields is not None and f.name not in fields and not f.has_default():
             continue
+
         if exclude and f.name in exclude:
             continue
         # Leave defaults for fields that aren't in POST data, except for
         # checkbox inputs because they don't appear in POST data if not checked.
-        if (f.has_default() and
-                form[f.name].field.widget.value_omitted_from_data(form.data, form.files, form.add_prefix(f.name))):
+
+
+        if (f.name in form.fields and f.has_default() and
+                form[f.name].field.widget.value_omitted_from_data(form.data, form.files, form.add_prefix(f.name))
+                and f.name not in form.changed_data):
             continue
+
+
         # Defer saving file-type fields until after the other fields, so a
         # callable upload_to can use the values from other fields.
         if isinstance(f, models.FileField):
@@ -83,8 +90,10 @@ def model_to_dict(instance, fields=None, exclude=None):
     for f in chain(opts.concrete_fields, opts.private_fields, opts.many_to_many):
         if not getattr(f, 'editable', False):
             continue
-        if fields is not None and f.name not in fields:
+
+        if fields is not None and f.name not in fields and not f.has_default():
             continue
+
         if exclude and f.name in exclude:
             continue
         data[f.name] = f.value_from_object(instance)
@@ -1355,4 +1364,4 @@ def modelform_defines_fields(form_class):
     return hasattr(form_class, '_meta') and (
         form_class._meta.fields is not None or
         form_class._meta.exclude is not None
-    )
+    )
