--- a/sympy/sets/sets.py
+++ b/sympy/sets/sets.py
@@ -861,6 +861,12 @@
 
     def __len__(self):
         return reduce(lambda a, b: a*b, (len(s) for s in self.args))
+
+    def _eval_is_subset(self, other):
+        if self.is_finite_set and self.is_iterable:
+            from sympy.core.containers import Tuple
+            return fuzzy_and(other._contains(Tuple(*e)) for e in self)
+        return None
 
     def __bool__(self):
         return all([bool(s) for s in self.sets])
