diff --git a/django/db/models/query.py b/django/db/models/query.py
index 1544beb613..ffe77dbae9 100644
--- a/django/db/models/query.py
+++ b/django/db/models/query.py
@@ -1224,11 +1224,20 @@ class QuerySet:
     def _clone(self):
         """
         Return a copy of the current QuerySet. A lightweight alternative
         to deepcopy().
         """
-        c = self.__class__(model=self.model, query=self.query.chain(), using=self._db, hints=self._hints)
+        query = self.query
+        # For combinators (e.g. union) we need a deep copy to avoid issues
+        # with the order_by columns being interpreted as references to
+        # columns in the select clause.
+        if query.combinator:
+            import copy
+            query = copy.deepcopy(query)
+        else:
+            query = query.chain()
+        c = self.__class__(model=self.model, query=query, using=self._db, hints=self._hints)
         c._sticky_filter = self._sticky_filter
         c._for_write = self._for_write
         c._prefetch_related_lookups = self._prefetch_related_lookups[:]
         c._known_related_objects = self._known_related_objects
         c._iterable_class = self._iterable_class
