diff --git a/django/db/backends/ddl_references.py b/django/db/backends/ddl_references.py
index c06386a2fa..ffa8304dce 100644
--- a/django/db/backends/ddl_references.py
+++ b/django/db/backends/ddl_references.py
@@ -53,6 +53,13 @@ class Table(Reference):
         if self.table == old_table:
             self.table = new_table
 
+    def __eq__(self, other):
+        if isinstance(other, Table):
+            return self.table == other.table
+        if isinstance(other, str):
+            return self.table == other
+        return NotImplemented
+
     def __str__(self):
         return self.quote_name(self.table)
 
