diff --git a/django/db/migrations/operations/models.py b/django/db/migrations/operations/models.py
index b086232..fd381ca 100644
--- a/django/db/migrations/operations/models.py
+++ b/django/db/migrations/operations/models.py
@@ -961,10 +961,16 @@ class RenameIndex(IndexOperation):
             from_model_state = from_state.models[app_label, self.model_name_lower]
             old_index = from_model_state.get_index_by_name(self.old_name)
 
+
         to_model_state = to_state.models[app_label, self.model_name_lower]
         new_index = to_model_state.get_index_by_name(self.new_name)
+        # If the current and target index names are identical (case‑insensitive),
+        # there's nothing to do.
+        if old_index.name.lower() == new_index.name.lower():
+            return
         schema_editor.rename_index(model, old_index, new_index)
 
+
     def database_backwards(self, app_label, schema_editor, from_state, to_state):
         if self.old_fields:
             # Backward operation with unnamed index is a no-op.
