diff --git a/sympy/polys/polytools.py b/sympy/polys/polytools.py
index d5ac051f63..86b18f9f3a 100644
--- a/sympy/polys/polytools.py
+++ b/sympy/polys/polytools.py
@@ -104,10 +104,11 @@ class Poly(Expr):
 
     __slots__ = ['rep', 'gens']
 
     is_commutative = True
     is_Poly = True
+    _op_priority = 20.0  # Higher than the default (10.0) to ensure Poly's methods are called
 
     def __new__(cls, rep, *gens, **args):
         """Create a new polynomial instance out of something useful. """
         opt = options.build_options(gens, args)
 
@@ -4051,11 +4052,14 @@ def __mul__(f, g):
     def __rmul__(f, g):
         if not g.is_Poly:
             try:
                 g = f.__class__(g, *f.gens)
             except PolynomialError:
-                return g*f.as_expr()
+                try:
+                    return f.__class__(g*f.as_expr(), *f.gens)
+                except PolynomialError:
+                    return g*f.as_expr()
 
         return g.mul(f)
 
     @_sympifyit('n', NotImplemented)
     def __pow__(f, n):
