diff --git a/xarray/core/variable.py b/xarray/core/variable.py
index 45553eb9..ac4630f5 100644
--- a/xarray/core/variable.py
+++ b/xarray/core/variable.py
@@ -215,12 +215,14 @@ def as_compatible_data(data, fastpath=False):
         data = np.datetime64(data.value, "ns")
 
     if isinstance(data, timedelta):
         data = np.timedelta64(getattr(data, "value", data), "ns")
 
-    # we don't want nested self-described arrays
-    data = getattr(data, "values", data)
+    # we don't want nested self-described arrays from pandas or xarray
+    if hasattr(data, "values") and (isinstance(data, (pd.DataFrame, pd.Series)) or 
+                                   type(data).__module__.startswith(('pandas.', 'xarray.'))):
+        data = data.values
 
     if isinstance(data, np.ma.MaskedArray):
         mask = np.ma.getmaskarray(data)
         if mask.any():
             dtype, fill_value = dtypes.maybe_promote(data.dtype)
