diff --git a/pylint/checkers/variables.py b/pylint/checkers/variables.py
index 8c0035f..f3f4b00 100644
--- a/pylint/checkers/variables.py
+++ b/pylint/checkers/variables.py
@@ -1822,13 +1822,20 @@ class VariablesChecker(BaseChecker):
 
     def _store_type_annotation_node(self, type_annotation):
         """Given a type annotation, store all the name nodes it refers to"""
+
         if isinstance(type_annotation, astroid.Name):
             self._type_annotation_names.append(type_annotation.name)
             return
 
+        if isinstance(type_annotation, astroid.Attribute):
+            for n in type_annotation.nodes_of_class(astroid.Name):
+                self._type_annotation_names.append(n.name)
+            return
+
         if not isinstance(type_annotation, astroid.Subscript):
             return
 
+
         if (
             isinstance(type_annotation.value, astroid.Attribute)
             and isinstance(type_annotation.value.expr, astroid.Name)
