diff --git a/sphinx/util/docfields.py b/sphinx/util/docfields.py
index 465c0ad..ea01999 100644
--- a/sphinx/util/docfields.py
+++ b/sphinx/util/docfields.py
@@ -297,8 +297,10 @@ class DocFieldTransformer:
 
             # also support syntax like ``:param type name:``
             if typedesc.is_typed:
+
                 try:
-                    argtype, argname = fieldarg.split(None, 1)
+                    argtype, argname = fieldarg.rsplit(None, 1)  # use right split
+
                 except ValueError:
                     pass
                 else:
