diff --git a/xarray/core/dataarray.py b/xarray/core/dataarray.py
index baab678..6c1108c 100644
--- a/xarray/core/dataarray.py
+++ b/xarray/core/dataarray.py
@@ -52,8 +52,10 @@ from .coordinates import (
 )
 from .dataset import Dataset, merge_indexes, split_indexes
 from .formatting import format_item
+
 from .indexes import Indexes, default_indexes
-from .options import OPTIONS
+from .options import OPTIONS, _get_keep_attrs
+
 from .utils import ReprObject, _check_inplace, either_dict_or_kwargs
 from .variable import (
     IndexVariable,
@@ -2872,10 +2874,15 @@ class DataArray(AbstractArray, DataWithCoords):
         numpy.nanpercentile, pandas.Series.quantile, Dataset.quantile
         """
 
+
         ds = self._to_temp_dataset().quantile(
             q, dim=dim, keep_attrs=keep_attrs, interpolation=interpolation
         )
-        return self._from_temp_dataset(ds)
+        out = self._from_temp_dataset(ds)
+        if _get_keep_attrs(keep_attrs):
+            out.attrs = self.attrs
+        return out
+
 
     def rank(
         self, dim: Hashable, pct: bool = False, keep_attrs: bool = None
