diff --git a/requests/adapters.py b/requests/adapters.py
index fe22ff45..d47134b4 100644
--- a/requests/adapters.py
+++ b/requests/adapters.py
@@ -535,4 +535,4 @@ class HTTPAdapter(BaseAdapter):
             else:
                 raise
 
-        return self.build_response(request, resp)
+        return self.build_response(request, resp)
diff --git a/requests/auth.py b/requests/auth.py
index eeface39..f6990ec2 100644
--- a/requests/auth.py
+++ b/requests/auth.py
@@ -302,4 +302,4 @@ class HTTPDigestAuth(AuthBase):
         ])
 
     def __ne__(self, other):
-        return not self == other
+        return not self == other
diff --git a/requests/compat.py b/requests/compat.py
index 029ae62a..f6a5c739 100644
--- a/requests/compat.py
+++ b/requests/compat.py
@@ -78,4 +78,4 @@ elif is_py3:
     bytes = bytes
     basestring = (str, bytes)
     numeric_types = (int, float)
-    integer_types = (int,)
+    integer_types = (int,)
diff --git a/requests/sessions.py b/requests/sessions.py
index 3f59cab9..594e72b9 100644
--- a/requests/sessions.py
+++ b/requests/sessions.py
@@ -768,4 +768,4 @@ def session():
 
     :rtype: Session
     """
-    return Session()
+    return Session()
diff --git a/requests/utils.py b/requests/utils.py
index 1c2ae4e0..b65bee61 100644
--- a/requests/utils.py
+++ b/requests/utils.py
@@ -20,10 +20,13 @@ import tempfile
 import warnings
 import zipfile
 from collections import OrderedDict
+
+
 from urllib3.util import make_headers
 from urllib3.util import parse_url
 
 from .__version__ import __version__
+
 from . import certs
 # to_native_string is unused here, but imported here for backwards compatibility
 from ._internal_utils import to_native_string
@@ -963,25 +966,25 @@ def prepend_scheme_if_needed(url, new_scheme):
 
     :rtype: str
     """
+
+    if "://" in url:
+        return url
+
+
     parsed = parse_url(url)
     scheme, auth, host, port, path, query, fragment = parsed
-
-    # A defect in urlparse determines that there isn't a netloc present in some
-    # urls. We previously assumed parsing was overly cautious, and swapped the
-    # netloc and path. Due to a lack of tests on the original defect, this is
-    # maintained with parse_url for backwards compatibility.
     netloc = parsed.netloc
     if not netloc:
         netloc, path = path, netloc
-
     if scheme is None:
         scheme = new_scheme
     if path is None:
         path = ''
-
     return urlunparse((scheme, netloc, path, '', query, fragment))
 
 
+
+
 def get_auth_from_url(url):
     """Given a url with authentication components, extract them into a tuple of
     username,password.
@@ -1053,4 +1056,4 @@ def rewind_body(prepared_request):
             raise UnrewindableBodyError("An error occurred when rewinding request "
                                         "body for redirect.")
     else:
-        raise UnrewindableBodyError("Unable to rewind request body for redirect.")
+        raise UnrewindableBodyError("Unable to rewind request body for redirect.")
