--- a/sklearn/tree/export.py
+++ b/sklearn/tree/export.py
@@ -889,10 +889,7 @@
     else:
         value_fmt = "{}{} value: {}\n"
 
-    if feature_names:
-        feature_names_ = [feature_names[i] for i in tree_.feature]
-    else:
-        feature_names_ = ["feature_{}".format(i) for i in tree_.feature]
+    # feature_names_ will be computed on-demand for internal nodes only
 
     export_text.report = ""
 
@@ -928,7 +925,10 @@
             info_fmt_right = info_fmt
 
             if tree_.feature[node] != _tree.TREE_UNDEFINED:
-                name = feature_names_[node]
+                if feature_names:
+                    name = feature_names[tree_.feature[node]]
+                else:
+                    name = "feature_{}".format(tree_.feature[node])
                 threshold = tree_.threshold[node]
                 threshold = "{1:.{0}f}".format(decimals, threshold)
                 export_text.report += right_child_fmt.format(indent,
