--- a/sphinx/extension.py
+++ b/sphinx/extension.py
@@ -9,6 +9,8 @@
 """
 
 from typing import TYPE_CHECKING, Any, Dict
+
+from packaging.version import InvalidVersion, Version
 
 from sphinx.config import Config
 from sphinx.errors import VersionRequirementError
@@ -51,11 +53,21 @@
                               'but it is not loaded.'), extname)
             continue
 
-        if extension.version == 'unknown version' or reqversion > extension.version:
+        if extension.version == 'unknown version':
             raise VersionRequirementError(__('This project needs the extension %s at least in '
                                              'version %s and therefore cannot be built with '
                                              'the loaded version (%s).') %
                                           (extname, reqversion, extension.version))
+        
+        try:
+            if Version(reqversion) > Version(extension.version):
+                raise VersionRequirementError(__('This project needs the extension %s at least in '
+                                                 'version %s and therefore cannot be built with '
+                                                 'the loaded version (%s).') %
+                                              (extname, reqversion, extension.version))
+        except InvalidVersion as exc:
+            logger.warning(__('Failed to check version requirement for %s (%s) due to invalid version string: %s'), 
+                           extname, reqversion, exc)
 
 
 def setup(app: "Sphinx") -> Dict[str, Any]:
