--- a/sphinx/ext/autodoc/__init__.py
+++ b/sphinx/ext/autodoc/__init__.py
@@ -121,6 +121,16 @@
     directives.flag(), which returns None).
     """
     return True
+
+
+def private_members_option(arg: Any) -> Union[object, bool, List[str]]:
+    """Used to convert the :private-members: option to auto directives."""
+    if arg is None:
+        return True
+    elif arg is True or arg is False:
+        return arg
+    else:
+        return [x.strip() for x in arg.split(',') if x.strip()]
 
 
 def merge_special_members_option(options: Dict) -> None:
@@ -649,15 +659,22 @@
             elif (namespace, membername) in attr_docs:
                 if want_all and isprivate:
                     # ignore members whose name starts with _ by default
-                    keep = self.options.private_members
+                    if isinstance(self.options.private_members, list):
+                        keep = membername in self.options.private_members
+                    else:
+                        keep = self.options.private_members
                 else:
                     # keep documented attributes
                     keep = True
                 isattr = True
             elif want_all and isprivate:
                 # ignore members whose name starts with _ by default
-                keep = self.options.private_members and \
-                    (has_doc or self.options.undoc_members)
+                if isinstance(self.options.private_members, list):
+                    keep = membername in self.options.private_members and \
+                        (has_doc or self.options.undoc_members)
+                else:
+                    keep = self.options.private_members and \
+                        (has_doc or self.options.undoc_members)
             else:
                 if self.options.members is ALL and is_filtered_inherited_member(membername):
                     keep = False
@@ -859,7 +876,7 @@
         'show-inheritance': bool_option, 'synopsis': identity,
         'platform': identity, 'deprecated': bool_option,
         'member-order': member_order_option, 'exclude-members': members_set_option,
-        'private-members': bool_option, 'special-members': members_option,
+        'private-members': private_members_option, 'special-members': members_option,
         'imported-members': bool_option, 'ignore-module-all': bool_option
     }  # type: Dict[str, Callable]
 
@@ -1279,7 +1296,7 @@
         'noindex': bool_option, 'inherited-members': inherited_members_option,
         'show-inheritance': bool_option, 'member-order': member_order_option,
         'exclude-members': members_set_option,
-        'private-members': bool_option, 'special-members': members_option,
+        'private-members': private_members_option, 'special-members': members_option,
     }  # type: Dict[str, Callable]
 
     _signature_class = None  # type: Any
