diff --git a/django/db/models/base.py b/django/db/models/base.py
index 6b13734..5e8866e 100644
--- a/django/db/models/base.py
+++ b/django/db/models/base.py
@@ -1746,8 +1746,15 @@ class Model(metaclass=ModelBase):
                         _cls = fld.get_path_info()[-1].to_opts.model
                     else:
                         _cls = None
+
                 except (FieldDoesNotExist, AttributeError):
-                    if fld is None or fld.get_transform(part) is None:
+                    has_transform = fld is not None and fld.get_transform(part) is not None
+                    try:
+                        has_lookup = fld is not None and fld.get_lookup(part) is not None
+                    except FieldError:
+                        has_lookup = False
+                    if not has_transform and not has_lookup:
+
                         errors.append(
                             checks.Error(
                                 "'ordering' refers to the nonexistent field, "
