diff --git a/django/core/management/commands/runserver.py b/django/core/management/commands/runserver.py
index c4f663b..9c2cea3 100644
--- a/django/core/management/commands/runserver.py
+++ b/django/core/management/commands/runserver.py
@@ -107,9 +107,14 @@ class Command(BaseCommand):
                     raise CommandError('"%s" is not a valid IPv6 address.' % self.addr)
         if not self.addr:
             self.addr = self.default_addr_ipv6 if self.use_ipv6 else self.default_addr
+
             self._raw_ipv6 = self.use_ipv6
+        # Treat the shorthand wildcard address "0" as "0.0.0.0".
+        if self.addr == "0" and not self.use_ipv6:
+            self.addr = "0.0.0.0"
         self.run(**options)
 
+
     def run(self, **options):
         """Run the server, using the autoreloader if needed."""
         use_reloader = options["use_reloader"]
@@ -180,4 +185,4 @@ class Command(BaseCommand):
         except KeyboardInterrupt:
             if shutdown_message:
                 self.stdout.write(shutdown_message)
-            sys.exit(0)
+            sys.exit(0)
