diff --git a/django/db/migrations/serializer.py b/django/db/migrations/serializer.py
index 27b5cbd379..eeb0969de6 100644
--- a/django/db/migrations/serializer.py
+++ b/django/db/migrations/serializer.py
@@ -120,11 +120,22 @@ class EnumSerializer(BaseSerializer):
     def serialize(self):
         enum_class = self.value.__class__
         module = enum_class.__module__
+
+
+        if isinstance(self.value.value, Promise):
+            # name-based lookup for lazy translations
+            n_string, n_imports = serializer_factory(self.value.name).serialize()
+            imports = {'import %s' % module, *n_imports}
+            return "%s.%s[%s]" % (module, enum_class.__name__, n_string), imports
+
         v_string, v_imports = serializer_factory(self.value.value).serialize()
         imports = {'import %s' % module, *v_imports}
         return "%s.%s(%s)" % (module, enum_class.__name__, v_string), imports
 
 
+
+
+
 class FloatSerializer(BaseSimpleSerializer):
     def serialize(self):
         if math.isnan(self.value) or math.isinf(self.value):
@@ -336,4 +347,4 @@ def serializer_factory(value):
         "Cannot serialize: %r\nThere are some values Django cannot serialize into "
         "migration files.\nFor more, see https://docs.djangoproject.com/en/%s/"
         "topics/migrations/#migration-serializing" % (value, get_docs_version())
-    )
+    )
diff --git a/django/db/migrations/writer.py b/django/db/migrations/writer.py
index 4918261fb0..36f0ec3e60 100644
--- a/django/db/migrations/writer.py
+++ b/django/db/migrations/writer.py
@@ -297,4 +297,4 @@ class Migration(migrations.Migration):
     operations = [
 %(operations)s\
     ]
-"""
+"""
diff --git a/django/db/models/enums.py b/django/db/models/enums.py
index bbe362a6ab..2924da38f2 100644
--- a/django/db/models/enums.py
+++ b/django/db/models/enums.py
@@ -72,4 +72,4 @@ class TextChoices(str, Choices):
     """Class for creating enumerated string choices."""
 
     def _generate_next_value_(name, start, count, last_values):
-        return name
+        return name
