diff --git a/django/db/models/enums.py b/django/db/models/enums.py
index 2924da3..7eaff82 100644
--- a/django/db/models/enums.py
+++ b/django/db/models/enums.py
@@ -25,8 +25,15 @@ class ChoicesMeta(enum.EnumMeta):
             # Use dict.__setitem__() to suppress defenses against double
             # assignment in enum's classdict.
             dict.__setitem__(classdict, key, value)
+
         cls = super().__new__(metacls, classname, bases, classdict)
+        # Ensure enum members convert to their primitive value when cast to str.
+        if '__str__' not in cls.__dict__:
+            def __str__(self):
+                return str(self.value)
+            cls.__str__ = __str__
         cls._value2label_map_ = dict(zip(cls._value2member_map_, labels))
+
         # Add a label property to instances of enum which uses the enum member
         # that is passed in as "self" as the value to use when looking up the
         # label in the choices.
@@ -72,4 +79,4 @@ class TextChoices(str, Choices):
     """Class for creating enumerated string choices."""
 
     def _generate_next_value_(name, start, count, last_values):
-        return name
+        return name
