diff --git a/django/contrib/auth/base_user.py b/django/contrib/auth/base_user.py
index 718fbb5..d3b1df6 100644
--- a/django/contrib/auth/base_user.py
+++ b/django/contrib/auth/base_user.py
@@ -135,13 +135,22 @@ class AbstractBaseUser(models.Model):
         """
         Return an HMAC of the password field.
         """
+
+        from django.conf import settings
         key_salt = "django.contrib.auth.models.AbstractBaseUser.get_session_auth_hash"
+        secret = (
+            settings.SECRET_KEY_FALLBACKS[0]
+            if settings.SECRET_KEY_FALLBACKS
+            else settings.SECRET_KEY
+        )
         return salted_hmac(
             key_salt,
             self.password,
+            secret=secret,
             algorithm="sha256",
         ).hexdigest()
 
+
     @classmethod
     def get_email_field_name(cls):
         try:
