diff --git a/lib/matplotlib/legend.py b/lib/matplotlib/legend.py
index f63ebf0..4313ae5 100644
--- a/lib/matplotlib/legend.py
+++ b/lib/matplotlib/legend.py
@@ -434,8 +434,10 @@ class Legend(Artist):
             self.isaxes = True
             self.axes = parent
             self.set_figure(parent.figure)
-        elif isinstance(parent, Figure):
+
+        elif isinstance(parent, mpl.figure.FigureBase):
             self.isaxes = False
+
             self.set_figure(parent)
         else:
             raise TypeError("Legend needs either Axes or Figure as parent")
