diff --git a/lib/matplotlib/dates.py b/lib/matplotlib/dates.py
index cc65fc1..eee20d0 100644
--- a/lib/matplotlib/dates.py
+++ b/lib/matplotlib/dates.py
@@ -805,8 +805,14 @@ class ConciseDateFormatter(ticker.Formatter):
                 # set to the most precise (6: microseconds doesn't exist...)
                 level = 5
 
+
         # level is the basic level we will label at.
         # now loop through and decide the actual ticklabels
+        # Re-enable the offset if we are labelling months and none of the
+        # ticks correspond to January, so the year is otherwise invisible.
+        if level == 1 and not show_offset and 1 not in tickdate[:, 1]:
+            show_offset = True
+
         zerovals = [0, 1, 1, 0, 0, 0, 0]
         labels = [''] * len(tickdate)
         for nn in range(len(tickdate)):
