diff --git a/lib/matplotlib/colors.py b/lib/matplotlib/colors.py
index bd73e56..d33393d 100644
--- a/lib/matplotlib/colors.py
+++ b/lib/matplotlib/colors.py
@@ -725,8 +725,16 @@ class Colormap:
                 # Avoid converting large positive values to negative integers.
                 np.clip(xa, -1, self.N, out=xa)
                 xa = xa.astype(int)
+
+        # Ensure integer arrays can accommodate sentinel values.
+        if xa.dtype.kind in {"i", "u"}:
+            # Promote if unsigned *or* unable to hold the largest sentinel.
+            if (xa.dtype.kind == "u"
+                    or not np.can_cast(self._i_bad, xa.dtype, casting="safe")):
+                xa = xa.astype(np.intp, copy=False)
         # Set the over-range indices before the under-range;
         # otherwise the under-range values get converted to over-range.
+
         xa[xa > self.N - 1] = self._i_over
         xa[xa < 0] = self._i_under
         xa[mask_bad] = self._i_bad
