diff --git a/lib/matplotlib/figure.py b/lib/matplotlib/figure.py
index eeb18de..5947c7e 100644
--- a/lib/matplotlib/figure.py
+++ b/lib/matplotlib/figure.py
@@ -1581,10 +1581,14 @@ default: %(va)s
             relative height of ``height_ratios[i] / sum(height_ratios)``.
             If not given, all rows will have the same height.
         """
+
         gs = GridSpec(nrows=nrows, ncols=ncols, figure=self,
                       wspace=wspace, hspace=hspace,
                       width_ratios=width_ratios,
                       height_ratios=height_ratios)
+        if wspace is not None or hspace is not None:
+            gs.update(wspace=wspace, hspace=hspace)
+
 
         sfarr = np.empty((nrows, ncols), dtype=object)
         for i in range(ncols):
