diff --git a/src/flask/blueprints.py b/src/flask/blueprints.py
index 678f5f0..f4dfe3f 100644
--- a/src/flask/blueprints.py
+++ b/src/flask/blueprints.py
@@ -188,11 +188,16 @@ class Blueprint(Scaffold):
             static_url_path=static_url_path,
             template_folder=template_folder,
             root_path=root_path,
+
         )
 
+        if not name or name.strip() == "":
+            raise ValueError("'name' must be a non-empty string.")
+
         if "." in name:
             raise ValueError("'name' may not contain a dot '.' character.")
 
+
         self.name = name
         self.url_prefix = url_prefix
         self.subdomain = subdomain
