diff --git a/sklearn/model_selection/_split.py b/sklearn/model_selection/_split.py
index 1bd5697..d6726d5 100644
--- a/sklearn/model_selection/_split.py
+++ b/sklearn/model_selection/_split.py
@@ -619,9 +619,11 @@ class StratifiedKFold(_BaseKFold):
             n_splits = 3
         super().__init__(n_splits, shuffle, random_state)
 
+
     def _make_test_folds(self, X, y=None):
-        rng = self.random_state
+        rng = check_random_state(self.random_state) if self.shuffle else self.random_state
         y = np.asarray(y)
+
         type_of_target_y = type_of_target(y)
         allowed_target_types = ('binary', 'multiclass')
         if type_of_target_y not in allowed_target_types:
