diff --git a/astropy/units/format/cds.py b/astropy/units/format/cds.py
index 761c4b8..fc985f6 100644
--- a/astropy/units/format/cds.py
+++ b/astropy/units/format/cds.py
@@ -179,11 +179,14 @@ class CDS(Base):
             else:
                 p[0] = p[1]
 
+
         def p_division_of_units(p):
             """
             division_of_units : DIVISION unit_expression
-                              | unit_expression DIVISION combined_units
+                              | division_of_units DIVISION unit_expression
+                              | unit_expression DIVISION unit_expression
             """
+
             if len(p) == 3:
                 p[0] = p[2] ** -1
             else:
@@ -355,4 +358,4 @@ class CDS(Base):
         elif isinstance(unit, core.NamedUnit):
             s = cls._get_unit_name(unit)
 
-        return s
+        return s
