diff --git a/django/db/models/base.py b/django/db/models/base.py
index 7468862..0b56560 100644
--- a/django/db/models/base.py
+++ b/django/db/models/base.py
@@ -1297,11 +1297,14 @@ class Model(metaclass=ModelBase):
 
     @classmethod
     def _check_default_pk(cls):
+
         if (
             cls._meta.pk.auto_created and
             not settings.is_overridden('DEFAULT_AUTO_FIELD') and
-            not cls._meta.app_config._is_default_auto_field_overridden
+            not cls._meta.app_config._is_default_auto_field_overridden and
+            not (getattr(cls._meta.pk, 'remote_field', None) and cls._meta.pk.remote_field.parent_link)
         ):
+
             return [
                 checks.Warning(
                     f"Auto-created primary key used when not defining a "
@@ -2136,4 +2139,4 @@ def model_unpickle(model_id):
     return model.__new__(model)
 
 
-model_unpickle.__safe_for_unpickle__ = True
+model_unpickle.__safe_for_unpickle__ = True
