diff --git a/xarray/core/computation.py b/xarray/core/computation.py
index 9fe93c88..542719d8 100644
--- a/xarray/core/computation.py
+++ b/xarray/core/computation.py
@@ -1809,6 +1809,7 @@ def where(cond, x, y):
         equivalent methods
     """
     # alignment for three arguments is complicated, so don't support it yet
+
     return apply_ufunc(
         duck_array_ops.where,
         cond,
@@ -1816,10 +1817,14 @@ def where(cond, x, y):
         y,
         join="exact",
         dataset_join="exact",
+
         dask="allowed",
+        keep_attrs="no_conflicts",
+
     )
 
 
+
 def polyval(coord, coeffs, degree_dim="degree"):
     """Evaluate a polynomial at specific values
 
@@ -1970,4 +1975,4 @@ def unify_chunks(*objects: T_Xarray) -> Tuple[T_Xarray, ...]:
                 ds._variables[k] = v.copy(data=next(dask_data_iter))
         out.append(obj._from_temp_dataset(ds) if isinstance(obj, DataArray) else ds)
 
-    return tuple(out)
+    return tuple(out)
diff --git a/xarray/core/dataarray.py b/xarray/core/dataarray.py
index 7f29d3b6..d07d6e51 100644
--- a/xarray/core/dataarray.py
+++ b/xarray/core/dataarray.py
@@ -4821,4 +4821,4 @@ class DataArray(AbstractArray, DataWithCoords, DataArrayArithmetic):
 
     # this needs to be at the end, or mypy will confuse with `str`
     # https://mypy.readthedocs.io/en/latest/common_issues.html#dealing-with-conflicting-names
-    str = utils.UncachedAccessor(StringAccessor)
+    str = utils.UncachedAccessor(StringAccessor)
diff --git a/xarray/core/duck_array_ops.py b/xarray/core/duck_array_ops.py
index 5b0d9a4f..e47ca902 100644
--- a/xarray/core/duck_array_ops.py
+++ b/xarray/core/duck_array_ops.py
@@ -650,4 +650,4 @@ def push(array, n, axis):
     if is_duck_dask_array(array):
         return dask_array_ops.push(array, n, axis)
     else:
-        return push(array, n, axis)
+        return push(array, n, axis)
diff --git a/xarray/core/variable.py b/xarray/core/variable.py
index 58aeceed..f1ee1d02 100644
--- a/xarray/core/variable.py
+++ b/xarray/core/variable.py
@@ -2996,4 +2996,4 @@ def assert_unique_multiindex_level_names(variables):
                 raise ValueError(
                     "conflicting level / dimension names. {} "
                     "already exists as a level name.".format(d)
-                )
+                )
