diff --git a/django/db/models/sql/compiler.py b/django/db/models/sql/compiler.py
index be54ccd..95f9328 100644
--- a/django/db/models/sql/compiler.py
+++ b/django/db/models/sql/compiler.py
@@ -61,6 +61,7 @@ class SQLCompiler:
         return extra_select, order_by, group_by
 
     def get_group_by(self, select, order_by):
+
         """
         Return a list of 2-tuples of form (sql, params).
 
@@ -68,7 +69,11 @@ class SQLCompiler:
         to describe in other words than "if it passes the test suite,
         then it is correct".
         """
+        # Guard against default Meta.ordering affecting GROUP BY.
+        if self._meta_ordering is not None and not self.query.order_by:
+            order_by = []
         # Some examples:
+
         #     SomeModel.objects.annotate(Count('somecol'))
         #     GROUP BY: all fields of the model
         #
