diff --git a/django/db/models/lookups.py b/django/db/models/lookups.py
index 01b769d..49f72df 100644
--- a/django/db/models/lookups.py
+++ b/django/db/models/lookups.py
@@ -81,9 +81,13 @@ class Lookup:
             lhs = lhs.resolve_expression(compiler.query)
         return compiler.compile(lhs)
 
+
     def process_rhs(self, compiler, connection):
         value = self.rhs
+        if hasattr(value, 'resolve_expression'):
+            value = value.resolve_expression(compiler.query)
         if self.bilateral_transforms:
+
             if self.rhs_is_direct_value():
                 # Do not call get_db_prep_lookup here as the value will be
                 # transformed before being used for lookup
