diff --git a/sphinx/domains/std.py b/sphinx/domains/std.py
index c7bf7da..4e84472 100644
--- a/sphinx/domains/std.py
+++ b/sphinx/domains/std.py
@@ -652,10 +652,12 @@ class StandardDomain(Domain):
 
         .. versionadded:: 3.0
         """
-        if (objtype, name) in self.objects:
+
+        if objtype != 'term' and (objtype, name) in self.objects:
             docname = self.objects[objtype, name][0]
             logger.warning(__('duplicate %s description of %s, other instance in %s'),
                            objtype, name, docname, location=location)
+
         self.objects[objtype, name] = (self.env.docname, labelid)
 
     def add_object(self, objtype: str, name: str, docname: str, labelid: str) -> None:
