diff --git a/django/db/migrations/serializer.py b/django/db/migrations/serializer.py
index 941a5ed..d6f8e5b 100644
--- a/django/db/migrations/serializer.py
+++ b/django/db/migrations/serializer.py
@@ -166,9 +166,11 @@ class FunctionTypeSerializer(BaseSerializer):
         if getattr(self.value, "__self__", None) and isinstance(
             self.value.__self__, type
         ):
+
             klass = self.value.__self__
             module = klass.__module__
-            return "%s.%s.%s" % (module, klass.__name__, self.value.__name__), {
+            return "%s.%s.%s" % (module, klass.__qualname__, self.value.__name__), {
+
                 "import %s" % module
             }
         # Further error checking
