diff --git a/django/forms/widgets.py b/django/forms/widgets.py
index dc396a1..11359a8 100644
--- a/django/forms/widgets.py
+++ b/django/forms/widgets.py
@@ -520,13 +520,17 @@ class CheckboxInput(Input):
             return
         return str(value)
 
+
     def get_context(self, name, value, attrs):
         if self.check_test(value):
             if attrs is None:
                 attrs = {}
+            else:
+                attrs = attrs.copy()
             attrs['checked'] = True
         return super().get_context(name, value, attrs)
 
+
     def value_from_datadict(self, data, files, name):
         if name not in data:
             # A missing value means False because HTML form submission does not
