diff --git a/django/db/migrations/operations/models.py b/django/db/migrations/operations/models.py
index 165c6f5..6d469cb 100644
--- a/django/db/migrations/operations/models.py
+++ b/django/db/migrations/operations/models.py
@@ -319,13 +319,23 @@ class RenameModel(ModelOperation):
     def database_forwards(self, app_label, schema_editor, from_state, to_state):
         new_model = to_state.apps.get_model(app_label, self.new_name)
         if self.allow_migrate_model(schema_editor.connection.alias, new_model):
+
             old_model = from_state.apps.get_model(app_label, self.old_name)
+            # Short-circuit if underlying table name isn't changing.
+            old_table = old_model._meta.db_table
+            new_table = new_model._meta.db_table
+            if (old_table == new_table or
+                (schema_editor.connection.features.ignores_table_name_case and
+                 old_table.lower() == new_table.lower())):
+                # The underlying table name is unchanged; no database changes required.
+                return
             # Move the main table
             schema_editor.alter_db_table(
                 new_model,
-                old_model._meta.db_table,
-                new_model._meta.db_table,
+                old_table,
+                new_table,
             )
+
             # Alter the fields pointing to us
             for related_object in old_model._meta.related_objects:
                 if related_object.related_model == old_model:
