diff --git a/django/db/models/sql/query.py b/django/db/models/sql/query.py
index cc286ae..02266c6 100644
--- a/django/db/models/sql/query.py
+++ b/django/db/models/sql/query.py
@@ -1074,13 +1074,19 @@ class Query(BaseExpression):
             value = value.resolve_expression(
                 self, reuse=can_reuse, allow_joins=allow_joins,
             )
+
         elif isinstance(value, (list, tuple)):
             # The items of the iterable may be expressions and therefore need
             # to be resolved independently.
-            return type(value)(
+            resolved_values = [
                 self.resolve_lookup_value(sub_value, can_reuse, allow_joins)
                 for sub_value in value
-            )
+            ]
+            constructor = type(value)
+            if hasattr(constructor, '_make'):
+                return constructor._make(resolved_values)
+            return constructor(resolved_values)
+
         return value
 
     def solve_lookup_type(self, lookup):
