diff --git a/django/db/models/sql/query.py b/django/db/models/sql/query.py
index 1309ca5..6f10625 100644
--- a/django/db/models/sql/query.py
+++ b/django/db/models/sql/query.py
@@ -1665,8 +1665,11 @@ class Query(BaseExpression):
         if isinstance(filter_rhs, F):
             filter_expr = (filter_lhs, OuterRef(filter_rhs.name))
         # Generate the inner query.
+
         query = Query(self.model)
+        query._filtered_relations = {alias: fr.clone() for alias, fr in self._filtered_relations.items()}
         query.add_filter(filter_expr)
+
         query.clear_ordering(True)
         # Try to have as simple as possible subquery -> trim leading joins from
         # the subquery.
