diff --git a/django/contrib/auth/checks.py b/django/contrib/auth/checks.py
index 2983afb..44d7400 100644
--- a/django/contrib/auth/checks.py
+++ b/django/contrib/auth/checks.py
@@ -3,9 +3,12 @@ from types import MethodType
 
 from django.apps import apps
 from django.conf import settings
+
 from django.core import checks
 
 from .management import _get_builtin_permissions
+from django.db.models import UniqueConstraint
+
 
 
 def check_user_model(app_configs=None, **kwargs):
@@ -51,8 +54,20 @@ def check_user_model(app_configs=None, **kwargs):
             )
         )
 
+
     # Check that the username field is unique
-    if not cls._meta.get_field(cls.USERNAME_FIELD).unique:
+    has_unique_username = (
+        cls._meta.get_field(cls.USERNAME_FIELD).unique
+        or any(
+            isinstance(c, UniqueConstraint)
+            and c.condition is None
+            and len(c.fields) == 1
+            and c.fields[0] == cls.USERNAME_FIELD
+            for c in cls._meta.total_unique_constraints
+        )
+    )
+    if not has_unique_username:
+
         if (settings.AUTHENTICATION_BACKENDS ==
                 ['django.contrib.auth.backends.ModelBackend']):
             errors.append(
