diff --git a/django/template/engine.py b/django/template/engine.py
index db7b746..d3bf66c 100644
--- a/django/template/engine.py
+++ b/django/template/engine.py
@@ -47,10 +47,13 @@ class Engine:
                 )
         if libraries is None:
             libraries = {}
+
         if builtins is None:
             builtins = []
 
+        dirs = [d for d in dirs if d]
         self.dirs = dirs
+
         self.app_dirs = app_dirs
         self.autoescape = autoescape
         self.context_processors = context_processors
