diff --git a/django/forms/models.py b/django/forms/models.py
index 2db5c72..dc97ced 100644
--- a/django/forms/models.py
+++ b/django/forms/models.py
@@ -812,8 +812,10 @@ class BaseModelFormSet(BaseFormSet):
         return saved_instances
 
     def save_new_objects(self, commit=True):
+
         self.new_objects = []
-        for form in self.extra_forms:
+        for form in self.extra_forms[:self.extra]:
+
             if not form.has_changed():
                 continue
             # If someone has marked an add form for deletion, don't save the
