diff --git a/django/db/backends/ddl_references.py b/django/db/backends/ddl_references.py
index 2b834d7..abf5134 100644
--- a/django/db/backends/ddl_references.py
+++ b/django/db/backends/ddl_references.py
@@ -77,11 +77,15 @@ class TableColumns(Table):
 class Columns(TableColumns):
     """Hold a reference to one or many columns."""
 
+
     def __init__(self, table, columns, quote_name, col_suffixes=()):
         self.quote_name = quote_name
         self.col_suffixes = col_suffixes
+        if isinstance(table, Table):
+            table = table.table
         super().__init__(table, columns)
 
+
     def __str__(self):
         def col_str(column, idx):
             col = self.quote_name(column)
