diff --git a/requests/sessions.py b/requests/sessions.py
index e1a79e6..35e9357 100644
--- a/requests/sessions.py
+++ b/requests/sessions.py
@@ -16,8 +16,10 @@ from .compat import cookielib, OrderedDict, urljoin, urlparse, urlunparse
 from .cookies import cookiejar_from_dict, extract_cookies_to_jar, RequestsCookieJar
 from .models import Request, PreparedRequest
 from .hooks import default_hooks, dispatch_hook
-from .utils import to_key_val_list, default_headers
+
+from .utils import to_key_val_list, default_headers, to_native_string
 from .exceptions import TooManyRedirects, InvalidSchema
+
 from .structures import CaseInsensitiveDict
 
 from .adapters import HTTPAdapter
@@ -254,8 +256,10 @@ class Session(SessionRedirectMixin):
 
         p = PreparedRequest()
         p.prepare(
-            method=request.method.upper(),
+
+            method=to_native_string(request.method).upper(),
             url=request.url,
+
             files=request.files,
             data=request.data,
             headers=merge_setting(request.headers, self.headers, dict_class=CaseInsensitiveDict),
@@ -310,9 +314,11 @@ class Session(SessionRedirectMixin):
             If Tuple, ('cert', 'key') pair.
         """
         # Create the Request.
+
         req = Request(
-            method = method.upper(),
+            method = to_native_string(method).upper(),
             url = url,
+
             headers = headers,
             files = files,
             data = data or {},
