diff --git a/sphinx/builders/__init__.py b/sphinx/builders/__init__.py
index 796b7ad..cfc8ff5 100644
--- a/sphinx/builders/__init__.py
+++ b/sphinx/builders/__init__.py
@@ -219,9 +219,18 @@ class Builder:
         logger.info(bold(__('building [mo]: ')) + message)
         for catalog in status_iterator(catalogs, __('writing output... '), "darkgreen",
                                        len(catalogs), self.app.verbosity,
+
                                        stringify_func=cat2relpath):
             catalog.write_mo(self.config.language)
 
+        # Refresh translations to take newly compiled catalogs into account
+        import gettext as _gettext
+        _gettext._translations.clear()
+
+        from sphinx import locale as _locale
+        _locale.init(self.config.locale_dirs, self.config.language)
+
+
     def compile_all_catalogs(self) -> None:
         repo = CatalogRepository(self.srcdir, self.config.locale_dirs,
                                  self.config.language, self.config.source_encoding)
