diff --git a/sphinx/ext/autodoc/__init__.py b/sphinx/ext/autodoc/__init__.py
index f4e1e57..a7b7ab5 100644
--- a/sphinx/ext/autodoc/__init__.py
+++ b/sphinx/ext/autodoc/__init__.py
@@ -829,9 +829,11 @@ class Documenter:
             classes.sort(key=lambda cls: cls.priority)
             # give explicitly separated module name, so that members
             # of inner classes can be documented
+
             full_mname = self.modname + '::' + \
                 '.'.join(self.objpath + [mname])
-            documenter = classes[-1](self.directive, full_mname, self.indent)
+            documenter = classes[-1](self.directive, full_mname, self.indent + '   ')
+
             memberdocumenters.append((documenter, isattr))
 
         member_order = self.options.member_order or self.config.autodoc_member_order
