diff --git a/sphinx/ext/autodoc/__init__.py b/sphinx/ext/autodoc/__init__.py
index d85d79617..4f365c040 100644
--- a/sphinx/ext/autodoc/__init__.py
+++ b/sphinx/ext/autodoc/__init__.py
@@ -1074,10 +1074,17 @@ class ModuleDocumenter(Documenter):
     def get_object_members(self, want_all: bool) -> Tuple[bool, ObjectMembers]:
         members = self.get_module_members()
         if want_all:
-            if not self.__all__:
+
+
+            if self.__all__ == []:
+                # Explicitly empty public interface -- expose nothing
+                return False, []
+            if self.__all__ is None:
                 # for implicit module members, check __module__ to avoid
+
                 # documenting imported objects
                 return True, list(members.values())
+
             else:
                 for member in members.values():
                     if member.__name__ not in self.__all__:
@@ -2614,4 +2621,4 @@ def setup(app: Sphinx) -> Dict[str, Any]:
     app.setup_extension('sphinx.ext.autodoc.type_comment')
     app.setup_extension('sphinx.ext.autodoc.typehints')
 
-    return {'version': sphinx.__display_version__, 'parallel_read_safe': True}
+    return {'version': sphinx.__display_version__, 'parallel_read_safe': True}
diff --git a/sphinx/ext/autodoc/directive.py b/sphinx/ext/autodoc/directive.py
index 9a3428f5d..846a505b0 100644
--- a/sphinx/ext/autodoc/directive.py
+++ b/sphinx/ext/autodoc/directive.py
@@ -155,4 +155,4 @@ class AutodocDirective(SphinxDirective):
             self.state.document.settings.record_dependencies.add(fn)
 
         result = parse_generated_content(self.state, params.result, documenter)
-        return result
+        return result
diff --git a/sphinx/ext/autodoc/importer.py b/sphinx/ext/autodoc/importer.py
index b6adb7ceb..85c454d9d 100644
--- a/sphinx/ext/autodoc/importer.py
+++ b/sphinx/ext/autodoc/importer.py
@@ -348,4 +348,4 @@ deprecated_alias('sphinx.ext.autodoc.importer',
                      'MockFinder': 'sphinx.ext.autodoc.mock.MockFinder',
                      'MockLoader': 'sphinx.ext.autodoc.mock.MockLoader',
                      'mock': 'sphinx.ext.autodoc.mock.mock',
-                 })
+                 })
diff --git a/sphinx/ext/autodoc/type_comment.py b/sphinx/ext/autodoc/type_comment.py
index debbc1442..37fff21aa 100644
--- a/sphinx/ext/autodoc/type_comment.py
+++ b/sphinx/ext/autodoc/type_comment.py
@@ -136,4 +136,4 @@ def update_annotations_using_type_comments(app: Sphinx, obj: Any, bound_method:
 def setup(app: Sphinx) -> Dict[str, Any]:
     app.connect('autodoc-before-process-signature', update_annotations_using_type_comments)
 
-    return {'version': sphinx.__display_version__, 'parallel_read_safe': True}
+    return {'version': sphinx.__display_version__, 'parallel_read_safe': True}
