diff --git a/sympy/core/exprtools.py b/sympy/core/exprtools.py
index 8f86407..90432e1 100644
--- a/sympy/core/exprtools.py
+++ b/sympy/core/exprtools.py
@@ -1175,8 +1175,10 @@ def factor_terms(expr, radical=False, clear=False, fraction=False, sign=True):
         cont, p = expr.as_content_primitive(radical=radical, clear=clear)
         if p.is_Add:
             list_args = [do(a) for a in Add.make_args(p)]
+
             # get a common negative (if there) which gcd_terms does not remove
-            if all(a.as_coeff_Mul()[0] < 0 for a in list_args):
+            if all(a.as_coeff_Mul()[0].is_negative is True for a in list_args):
+
                 cont = -cont
                 list_args = [-a for a in list_args]
             # watch out for exp(-(x+2)) which gcd_terms will change to exp(-x-2)
diff --git a/sympy/matrices/matrices.py b/sympy/matrices/matrices.py
index 3216a40..34a3e6b 100644
--- a/sympy/matrices/matrices.py
+++ b/sympy/matrices/matrices.py
@@ -206,11 +206,13 @@ class MatrixDeterminant(MatrixCommon):
             tmp_mat = mat.extract(rows, cols)
 
             def entry(i, j):
+
                 ret = (pivot_val*tmp_mat[i, j + 1] - mat[pivot_pos, j + 1]*tmp_mat[i, 0]) / cumm
                 if not ret.is_Atom:
-                    cancel(ret)
+                    ret = cancel(ret)
                 return ret
 
+
             return sign*bareiss(self._new(mat.rows - 1, mat.cols - 1, entry), pivot_val)
 
         return cancel(bareiss(self))
