diff --git a/sympy/sets/handlers/intersection.py b/sympy/sets/handlers/intersection.py
index 99c7e51..ab3990d 100644
--- a/sympy/sets/handlers/intersection.py
+++ b/sympy/sets/handlers/intersection.py
@@ -1,10 +1,13 @@
 from sympy import (S, Dummy, Lambda, symbols, Interval, Intersection, Set,
                    EmptySet, FiniteSet, Union, ComplexRegion, Mul)
 from sympy.multipledispatch import dispatch
+
 from sympy.sets.conditionset import ConditionSet
 from sympy.sets.fancysets import (Integers, Naturals, Reals, Range,
     ImageSet, Rationals)
 from sympy.sets.sets import UniversalSet, imageset, ProductSet
+from sympy.solvers.solveset import solveset_real
+
 
 
 @dispatch(ConditionSet, ConditionSet)  # type: ignore # noqa:F811
