diff --git a/sympy/printing/pycode.py b/sympy/printing/pycode.py
index d7c4404..9354310 100644
--- a/sympy/printing/pycode.py
+++ b/sympy/printing/pycode.py
@@ -16,9 +16,13 @@ _kw = {
     'with', 'yield', 'None', 'False', 'nonlocal', 'True'
 }
 
+
 _known_functions = {
     'Abs': 'abs',
+    'Min': 'min',      # added to support sympy.Min
+    'Max': 'max',      # added to support sympy.Max
 }
+
 _known_functions_math = {
     'acos': 'acos',
     'acosh': 'acosh',
